// @dydxprotocol/v4-client-js/build/src/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubaccountInfo = exports.encodeJson = exports.NetworkOptimizer = exports.SocketClient = exports.FaucetClient = exports.ValidatorClient = exports.IndexerClient = exports.NobleClient = exports.CompositeClient = exports.SubaccountClient = exports.LocalWallet = exports.onboarding = exports.validation = exports.utils = exports.helpers = void 0;
// Types.
__exportStar(require("./types"), exports);
// Utility functions.
exports.helpers = __importStar(require("./lib/helpers"));
exports.utils = __importStar(require("./lib/utils"));
exports.validation = __importStar(require("./lib/validation"));
exports.onboarding = __importStar(require("./lib/onboarding"));
var local_wallet_1 = require("./clients/modules/local-wallet");
Object.defineProperty(exports, "LocalWallet", { enumerable: true, get: function () { return __importDefault(local_wallet_1).default; } });
var subaccount_1 = require("./clients/subaccount");
Object.defineProperty(exports, "SubaccountClient", { enumerable: true, get: function () { return subaccount_1.SubaccountInfo; } });
var composite_client_1 = require("./clients/composite-client");
Object.defineProperty(exports, "CompositeClient", { enumerable: true, get: function () { return composite_client_1.CompositeClient; } });
var noble_client_1 = require("./clients/noble-client");
Object.defineProperty(exports, "NobleClient", { enumerable: true, get: function () { return noble_client_1.NobleClient; } });
var indexer_client_1 = require("./clients/indexer-client");
Object.defineProperty(exports, "IndexerClient", { enumerable: true, get: function () { return indexer_client_1.IndexerClient; } });
var validator_client_1 = require("./clients/validator-client");
Object.defineProperty(exports, "ValidatorClient", { enumerable: true, get: function () { return validator_client_1.ValidatorClient; } });
var faucet_client_1 = require("./clients/faucet-client");
Object.defineProperty(exports, "FaucetClient", { enumerable: true, get: function () { return faucet_client_1.FaucetClient; } });
var socket_client_1 = require("./clients/socket-client");
Object.defineProperty(exports, "SocketClient", { enumerable: true, get: function () { return socket_client_1.SocketClient; } });
var network_optimizer_1 = require("./network_optimizer");
Object.defineProperty(exports, "NetworkOptimizer", { enumerable: true, get: function () { return network_optimizer_1.NetworkOptimizer; } });
var helpers_1 = require("./lib/helpers");
Object.defineProperty(exports, "encodeJson", { enumerable: true, get: function () { return helpers_1.encodeJson; } });
var subaccount_2 = require("./clients/subaccount");
Object.defineProperty(exports, "SubaccountInfo", { enumerable: true, get: function () { return subaccount_2.SubaccountInfo; } });
