// @dydxprotocol/v4-client-js/build/src/network_optimizer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkOptimizer = exports.isTruthy = void 0;
const indexer_client_1 = require("./clients/indexer-client");
const validator_client_1 = require("./clients/validator-client");
const helpers_1 = require("./lib/helpers");
const types_1 = require("./types");
class PingResponse {
    constructor(height) {
        this.height = height;
        this.responseTime = new Date();
    }
}
const isTruthy = (n) => Boolean(n);
exports.isTruthy = isTruthy;
class NetworkOptimizer {
    async validatorClients(endpointUrls, chainId) {
        return (await Promise.all(endpointUrls.map((endpointUrl) => validator_client_1.ValidatorClient.connect(new types_1.ValidatorConfig(endpointUrl, chainId, {
            CHAINTOKEN_DENOM: 'placeholder',
            CHAINTOKEN_DECIMALS: 18,
            USDC_DENOM: 'uusdc',
            USDC_DECIMALS: 6,
        })).catch((_) => undefined)))).filter(exports.isTruthy);
    }
    indexerClients(endpointUrls) {
        return endpointUrls
            .map((endpointUrl) => new indexer_client_1.IndexerClient(
        // socket is not used for finding optimal indexer, but required as a parameter to the config
        new types_1.IndexerConfig(endpointUrl, endpointUrl.replace('https://', 'wss://').replace('http://', 'ws://'))))
            .filter(exports.isTruthy);
    }
    async findOptimalNode(endpointUrls, chainId) {
        if (endpointUrls.length === 0) {
            const errorResponse = {
                error: {
                    message: 'No nodes provided',
                },
            };
            return (0, helpers_1.encodeJson)(errorResponse);
        }
        const clients = await this.validatorClients(endpointUrls, chainId);
        const responses = (await Promise.all(clients
            .map(async (client) => {
            const block = await client.get.latestBlock();
            const response = new PingResponse(block.header.height);
            return {
                endpoint: client.config.restEndpoint,
                height: response.height,
                time: response.responseTime.getTime(),
            };
        })
            .map((promise) => promise.catch((_) => undefined)))).filter(exports.isTruthy);
        if (responses.length === 0) {
            throw new Error('Could not connect to endpoints');
        }
        const maxHeight = Math.max(...responses.map(({ height }) => height));
        return (responses
            // Only consider nodes at `maxHeight` or `maxHeight - 1`
            .filter(({ height }) => height === maxHeight || height === maxHeight - 1)
            // Return the endpoint with the fastest response time
            .sort((a, b) => a.time - b.time)[0].endpoint);
    }
    async findOptimalIndexer(endpointUrls) {
        if (endpointUrls.length === 0) {
            const errorResponse = {
                error: {
                    message: 'No URL provided',
                },
            };
            return (0, helpers_1.encodeJson)(errorResponse);
        }
        const clients = this.indexerClients(endpointUrls);
        const responses = (await Promise.all(clients
            .map(async (client) => {
            const block = await client.utility.getHeight();
            const response = new PingResponse(+block.height);
            return {
                endpoint: client.config.restEndpoint,
                height: response.height,
                time: response.responseTime.getTime(),
            };
        })
            .map((promise) => promise.catch((_) => undefined)))).filter(exports.isTruthy);
        if (responses.length === 0) {
            throw new Error('Could not connect to endpoints');
        }
        const maxHeight = Math.max(...responses.map(({ height }) => height));
        return (responses
            // Only consider nodes at `maxHeight` or `maxHeight - 1`
            .filter(({ height }) => height === maxHeight || height === maxHeight - 1)
            // Return the endpoint with the fastest response time
            .sort((a, b) => a.time - b.time)[0].endpoint);
    }
}
exports.NetworkOptimizer = NetworkOptimizer;
