// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/govplus/tx.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgSlashValidatorResponse = exports.MsgSlashValidator = void 0;
const _m0 = __importStar(require("protobufjs/minimal"));
function createBaseMsgSlashValidator() {
    return {
        authority: "",
        validatorAddress: "",
        infractionHeight: 0,
        tokensAtInfractionHeight: new Uint8Array(),
        slashFactor: ""
    };
}
exports.MsgSlashValidator = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.validatorAddress !== "") {
            writer.uint32(18).string(message.validatorAddress);
        }
        if (message.infractionHeight !== 0) {
            writer.uint32(24).uint32(message.infractionHeight);
        }
        if (message.tokensAtInfractionHeight.length !== 0) {
            writer.uint32(34).bytes(message.tokensAtInfractionHeight);
        }
        if (message.slashFactor !== "") {
            writer.uint32(42).string(message.slashFactor);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSlashValidator();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.validatorAddress = reader.string();
                    break;
                case 3:
                    message.infractionHeight = reader.uint32();
                    break;
                case 4:
                    message.tokensAtInfractionHeight = reader.bytes();
                    break;
                case 5:
                    message.slashFactor = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e;
        const message = createBaseMsgSlashValidator();
        message.authority = (_a = object.authority) !== null && _a !== void 0 ? _a : "";
        message.validatorAddress = (_b = object.validatorAddress) !== null && _b !== void 0 ? _b : "";
        message.infractionHeight = (_c = object.infractionHeight) !== null && _c !== void 0 ? _c : 0;
        message.tokensAtInfractionHeight = (_d = object.tokensAtInfractionHeight) !== null && _d !== void 0 ? _d : new Uint8Array();
        message.slashFactor = (_e = object.slashFactor) !== null && _e !== void 0 ? _e : "";
        return message;
    }
};
function createBaseMsgSlashValidatorResponse() {
    return {};
}
exports.MsgSlashValidatorResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSlashValidatorResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgSlashValidatorResponse();
        return message;
    }
};
