// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/affiliates/affiliates.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AffiliateWhitelist_Tier = exports.AffiliateWhitelist = exports.AffiliateTiers_Tier = exports.AffiliateTiers = void 0;
const _m0 = __importStar(require("protobufjs/minimal"));
const helpers_1 = require("../../helpers");
function createBaseAffiliateTiers() {
    return {
        tiers: []
    };
}
exports.AffiliateTiers = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.tiers) {
            exports.AffiliateTiers_Tier.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAffiliateTiers();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.tiers.push(exports.AffiliateTiers_Tier.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseAffiliateTiers();
        message.tiers = ((_a = object.tiers) === null || _a === void 0 ? void 0 : _a.map(e => exports.AffiliateTiers_Tier.fromPartial(e))) || [];
        return message;
    }
};
function createBaseAffiliateTiers_Tier() {
    return {
        reqReferredVolumeQuoteQuantums: helpers_1.Long.UZERO,
        reqStakedWholeCoins: 0,
        takerFeeSharePpm: 0
    };
}
exports.AffiliateTiers_Tier = {
    encode(message, writer = _m0.Writer.create()) {
        if (!message.reqReferredVolumeQuoteQuantums.isZero()) {
            writer.uint32(8).uint64(message.reqReferredVolumeQuoteQuantums);
        }
        if (message.reqStakedWholeCoins !== 0) {
            writer.uint32(16).uint32(message.reqStakedWholeCoins);
        }
        if (message.takerFeeSharePpm !== 0) {
            writer.uint32(24).uint32(message.takerFeeSharePpm);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAffiliateTiers_Tier();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.reqReferredVolumeQuoteQuantums = reader.uint64();
                    break;
                case 2:
                    message.reqStakedWholeCoins = reader.uint32();
                    break;
                case 3:
                    message.takerFeeSharePpm = reader.uint32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseAffiliateTiers_Tier();
        message.reqReferredVolumeQuoteQuantums = object.reqReferredVolumeQuoteQuantums !== undefined && object.reqReferredVolumeQuoteQuantums !== null ? helpers_1.Long.fromValue(object.reqReferredVolumeQuoteQuantums) : helpers_1.Long.UZERO;
        message.reqStakedWholeCoins = (_a = object.reqStakedWholeCoins) !== null && _a !== void 0 ? _a : 0;
        message.takerFeeSharePpm = (_b = object.takerFeeSharePpm) !== null && _b !== void 0 ? _b : 0;
        return message;
    }
};
function createBaseAffiliateWhitelist() {
    return {
        tiers: []
    };
}
exports.AffiliateWhitelist = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.tiers) {
            exports.AffiliateWhitelist_Tier.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAffiliateWhitelist();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.tiers.push(exports.AffiliateWhitelist_Tier.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseAffiliateWhitelist();
        message.tiers = ((_a = object.tiers) === null || _a === void 0 ? void 0 : _a.map(e => exports.AffiliateWhitelist_Tier.fromPartial(e))) || [];
        return message;
    }
};
function createBaseAffiliateWhitelist_Tier() {
    return {
        addresses: [],
        takerFeeSharePpm: 0
    };
}
exports.AffiliateWhitelist_Tier = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.addresses) {
            writer.uint32(10).string(v);
        }
        if (message.takerFeeSharePpm !== 0) {
            writer.uint32(16).uint32(message.takerFeeSharePpm);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAffiliateWhitelist_Tier();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.addresses.push(reader.string());
                    break;
                case 2:
                    message.takerFeeSharePpm = reader.uint32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseAffiliateWhitelist_Tier();
        message.addresses = ((_a = object.addresses) === null || _a === void 0 ? void 0 : _a.map(e => e)) || [];
        message.takerFeeSharePpm = (_b = object.takerFeeSharePpm) !== null && _b !== void 0 ? _b : 0;
        return message;
    }
};
