// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/affiliates/query.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AffiliateWhitelistResponse = exports.AffiliateWhitelistRequest = exports.AllAffiliateTiersResponse = exports.AllAffiliateTiersRequest = exports.ReferredByResponse = exports.ReferredByRequest = exports.AffiliateInfoResponse = exports.AffiliateInfoRequest = void 0;
const affiliates_1 = require("./affiliates");
const _m0 = __importStar(require("protobufjs/minimal"));
function createBaseAffiliateInfoRequest() {
    return {
        address: ""
    };
}
exports.AffiliateInfoRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAffiliateInfoRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.address = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseAffiliateInfoRequest();
        message.address = (_a = object.address) !== null && _a !== void 0 ? _a : "";
        return message;
    }
};
function createBaseAffiliateInfoResponse() {
    return {
        isWhitelisted: false,
        tier: 0,
        feeSharePpm: 0,
        referredVolume: new Uint8Array(),
        stakedAmount: new Uint8Array()
    };
}
exports.AffiliateInfoResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.isWhitelisted === true) {
            writer.uint32(8).bool(message.isWhitelisted);
        }
        if (message.tier !== 0) {
            writer.uint32(16).uint32(message.tier);
        }
        if (message.feeSharePpm !== 0) {
            writer.uint32(24).uint32(message.feeSharePpm);
        }
        if (message.referredVolume.length !== 0) {
            writer.uint32(34).bytes(message.referredVolume);
        }
        if (message.stakedAmount.length !== 0) {
            writer.uint32(42).bytes(message.stakedAmount);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAffiliateInfoResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.isWhitelisted = reader.bool();
                    break;
                case 2:
                    message.tier = reader.uint32();
                    break;
                case 3:
                    message.feeSharePpm = reader.uint32();
                    break;
                case 4:
                    message.referredVolume = reader.bytes();
                    break;
                case 5:
                    message.stakedAmount = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e;
        const message = createBaseAffiliateInfoResponse();
        message.isWhitelisted = (_a = object.isWhitelisted) !== null && _a !== void 0 ? _a : false;
        message.tier = (_b = object.tier) !== null && _b !== void 0 ? _b : 0;
        message.feeSharePpm = (_c = object.feeSharePpm) !== null && _c !== void 0 ? _c : 0;
        message.referredVolume = (_d = object.referredVolume) !== null && _d !== void 0 ? _d : new Uint8Array();
        message.stakedAmount = (_e = object.stakedAmount) !== null && _e !== void 0 ? _e : new Uint8Array();
        return message;
    }
};
function createBaseReferredByRequest() {
    return {
        address: ""
    };
}
exports.ReferredByRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseReferredByRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.address = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseReferredByRequest();
        message.address = (_a = object.address) !== null && _a !== void 0 ? _a : "";
        return message;
    }
};
function createBaseReferredByResponse() {
    return {
        affiliateAddress: ""
    };
}
exports.ReferredByResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.affiliateAddress !== "") {
            writer.uint32(10).string(message.affiliateAddress);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseReferredByResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.affiliateAddress = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseReferredByResponse();
        message.affiliateAddress = (_a = object.affiliateAddress) !== null && _a !== void 0 ? _a : "";
        return message;
    }
};
function createBaseAllAffiliateTiersRequest() {
    return {};
}
exports.AllAffiliateTiersRequest = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAllAffiliateTiersRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseAllAffiliateTiersRequest();
        return message;
    }
};
function createBaseAllAffiliateTiersResponse() {
    return {
        tiers: undefined
    };
}
exports.AllAffiliateTiersResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.tiers !== undefined) {
            affiliates_1.AffiliateTiers.encode(message.tiers, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAllAffiliateTiersResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.tiers = affiliates_1.AffiliateTiers.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseAllAffiliateTiersResponse();
        message.tiers = object.tiers !== undefined && object.tiers !== null ? affiliates_1.AffiliateTiers.fromPartial(object.tiers) : undefined;
        return message;
    }
};
function createBaseAffiliateWhitelistRequest() {
    return {};
}
exports.AffiliateWhitelistRequest = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAffiliateWhitelistRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseAffiliateWhitelistRequest();
        return message;
    }
};
function createBaseAffiliateWhitelistResponse() {
    return {
        whitelist: undefined
    };
}
exports.AffiliateWhitelistResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.whitelist !== undefined) {
            affiliates_1.AffiliateWhitelist.encode(message.whitelist, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAffiliateWhitelistResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.whitelist = affiliates_1.AffiliateWhitelist.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseAffiliateWhitelistResponse();
        message.whitelist = object.whitelist !== undefined && object.whitelist !== null ? affiliates_1.AffiliateWhitelist.fromPartial(object.whitelist) : undefined;
        return message;
    }
};
