// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/affiliates/tx.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgUpdateAffiliateWhitelistResponse = exports.MsgUpdateAffiliateWhitelist = exports.MsgUpdateAffiliateTiersResponse = exports.MsgUpdateAffiliateTiers = exports.MsgRegisterAffiliateResponse = exports.MsgRegisterAffiliate = void 0;
const affiliates_1 = require("./affiliates");
const _m0 = __importStar(require("protobufjs/minimal"));
function createBaseMsgRegisterAffiliate() {
    return {
        referee: "",
        affiliate: ""
    };
}
exports.MsgRegisterAffiliate = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.referee !== "") {
            writer.uint32(10).string(message.referee);
        }
        if (message.affiliate !== "") {
            writer.uint32(18).string(message.affiliate);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgRegisterAffiliate();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.referee = reader.string();
                    break;
                case 2:
                    message.affiliate = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseMsgRegisterAffiliate();
        message.referee = (_a = object.referee) !== null && _a !== void 0 ? _a : "";
        message.affiliate = (_b = object.affiliate) !== null && _b !== void 0 ? _b : "";
        return message;
    }
};
function createBaseMsgRegisterAffiliateResponse() {
    return {};
}
exports.MsgRegisterAffiliateResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgRegisterAffiliateResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgRegisterAffiliateResponse();
        return message;
    }
};
function createBaseMsgUpdateAffiliateTiers() {
    return {
        authority: "",
        tiers: undefined
    };
}
exports.MsgUpdateAffiliateTiers = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.tiers !== undefined) {
            affiliates_1.AffiliateTiers.encode(message.tiers, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateAffiliateTiers();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.tiers = affiliates_1.AffiliateTiers.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseMsgUpdateAffiliateTiers();
        message.authority = (_a = object.authority) !== null && _a !== void 0 ? _a : "";
        message.tiers = object.tiers !== undefined && object.tiers !== null ? affiliates_1.AffiliateTiers.fromPartial(object.tiers) : undefined;
        return message;
    }
};
function createBaseMsgUpdateAffiliateTiersResponse() {
    return {};
}
exports.MsgUpdateAffiliateTiersResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateAffiliateTiersResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgUpdateAffiliateTiersResponse();
        return message;
    }
};
function createBaseMsgUpdateAffiliateWhitelist() {
    return {
        authority: "",
        whitelist: undefined
    };
}
exports.MsgUpdateAffiliateWhitelist = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.whitelist !== undefined) {
            affiliates_1.AffiliateWhitelist.encode(message.whitelist, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateAffiliateWhitelist();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.whitelist = affiliates_1.AffiliateWhitelist.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseMsgUpdateAffiliateWhitelist();
        message.authority = (_a = object.authority) !== null && _a !== void 0 ? _a : "";
        message.whitelist = object.whitelist !== undefined && object.whitelist !== null ? affiliates_1.AffiliateWhitelist.fromPartial(object.whitelist) : undefined;
        return message;
    }
};
function createBaseMsgUpdateAffiliateWhitelistResponse() {
    return {};
}
exports.MsgUpdateAffiliateWhitelistResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateAffiliateWhitelistResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgUpdateAffiliateWhitelistResponse();
        return message;
    }
};
