// @dydxprotocol/v4-client-js/build/src/clients/modules/composer.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Composer = void 0;
const clob_pair_1 = require("@dydxprotocol/v4-proto/src/codegen/dydxprotocol/clob/clob_pair");
const long_1 = __importDefault(require("long"));
const protobufjs_1 = __importDefault(require("protobufjs"));
const constants_1 = require("../constants");
const proto_includes_1 = require("./proto-includes");
protobufjs_1.default.util.Long = long_1.default;
protobufjs_1.default.configure();
class Composer {
    // ------------ x/clob ------------
    composeMsgPlaceOrder(address, subaccountNumber, clientId, clobPairId, orderFlags, goodTilBlock, goodTilBlockTime, side, quantums, subticks, timeInForce, reduceOnly, clientMetadata, conditionType = proto_includes_1.Order_ConditionType.CONDITION_TYPE_UNSPECIFIED, conditionalOrderTriggerSubticks = long_1.default.fromInt(0)) {
        this.validateGoodTilBlockAndTime(orderFlags, goodTilBlock, goodTilBlockTime);
        const subaccountId = {
            owner: address,
            number: subaccountNumber,
        };
        const orderId = {
            subaccountId,
            clientId,
            orderFlags,
            clobPairId,
        };
        const order = {
            orderId,
            side,
            quantums,
            subticks,
            goodTilBlock: goodTilBlock === 0 ? undefined : goodTilBlock,
            goodTilBlockTime: goodTilBlock === 0 ? goodTilBlockTime : undefined,
            timeInForce,
            reduceOnly,
            clientMetadata: clientMetadata !== null && clientMetadata !== void 0 ? clientMetadata : 0,
            conditionType,
            conditionalOrderTriggerSubticks,
        };
        const msg = {
            order,
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_PLACE_ORDER,
            value: msg,
        };
    }
    composeMsgCancelOrder(address, subaccountNumber, clientId, clobPairId, orderFlags, goodTilBlock, goodTilBlockTime) {
        this.validateGoodTilBlockAndTime(orderFlags, goodTilBlock, goodTilBlockTime);
        const subaccountId = {
            owner: address,
            number: subaccountNumber,
        };
        const orderId = {
            subaccountId,
            clientId,
            orderFlags,
            clobPairId,
        };
        const msg = {
            orderId,
            goodTilBlock: goodTilBlock === 0 ? undefined : goodTilBlock,
            goodTilBlockTime: goodTilBlock === 0 ? goodTilBlockTime : undefined,
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_CANCEL_ORDER,
            value: msg,
        };
    }
    composeMsgBatchCancelShortTermOrders(address, subaccountNumber, shortTermCancels, goodTilBlock) {
        const subaccountId = {
            owner: address,
            number: subaccountNumber,
        };
        const msg = {
            subaccountId,
            shortTermCancels,
            goodTilBlock,
        };
        return {
            typeUrl: constants_1.TYPE_URL_BATCH_CANCEL,
            value: msg,
        };
    }
    composeMsgCreateClobPair(clobId, perpetualId, quantumConversionExponent, stepBaseQuantums, subticksPerTick) {
        const msg = {
            // uses x/gov module account since creating the clob pair is a governance action.
            authority: constants_1.GOV_MODULE_ADDRESS,
            clobPair: {
                id: clobId,
                perpetualClobMetadata: {
                    perpetualId,
                },
                quantumConversionExponent,
                stepBaseQuantums,
                subticksPerTick,
                status: clob_pair_1.ClobPair_Status.STATUS_INITIALIZING,
            },
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_CREATE_CLOB_PAIR,
            value: msg,
        };
    }
    composeMsgUpdateClobPair(clobId, perpetualId, quantumConversionExponent, stepBaseQuantums, subticksPerTick) {
        const msg = {
            // uses x/delaymsg module account since updating the clob pair is a delayedmsg action.
            authority: constants_1.DELAYMSG_MODULE_ADDRESS,
            clobPair: {
                id: clobId,
                perpetualClobMetadata: {
                    perpetualId,
                },
                quantumConversionExponent,
                stepBaseQuantums,
                subticksPerTick,
                status: clob_pair_1.ClobPair_Status.STATUS_ACTIVE,
            },
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_UPDATE_CLOB_PAIR,
            value: msg,
        };
    }
    // ------------ x/sending ------------
    composeMsgTransfer(address, subaccountNumber, recipientAddress, recipientSubaccountNumber, assetId, amount) {
        const sender = {
            owner: address,
            number: subaccountNumber,
        };
        const recipient = {
            owner: recipientAddress,
            number: recipientSubaccountNumber,
        };
        const transfer = {
            sender,
            recipient,
            assetId,
            amount,
        };
        const msg = {
            transfer,
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_CREATE_TRANSFER,
            value: msg,
        };
    }
    composeMsgDepositToSubaccount(address, subaccountNumber, assetId, quantums) {
        const recipient = {
            owner: address,
            number: subaccountNumber,
        };
        const msg = {
            sender: address,
            recipient,
            assetId,
            quantums,
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_DEPOSIT_TO_SUBACCOUNT,
            value: msg,
        };
    }
    composeMsgWithdrawFromSubaccount(address, subaccountNumber, assetId, quantums, recipient = address) {
        const sender = {
            owner: address,
            number: subaccountNumber,
        };
        const msg = {
            sender,
            recipient,
            assetId,
            quantums,
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_WITHDRAW_FROM_SUBACCOUNT,
            value: msg,
        };
    }
    // ------------ x/bank ------------
    composeMsgSendToken(address, recipient, coinDenom, quantums) {
        const coin = {
            denom: coinDenom,
            amount: quantums,
        };
        const msg = {
            fromAddress: address,
            toAddress: recipient,
            amount: [coin],
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_SEND,
            value: msg,
        };
    }
    // ------------ x/prices ------------
    composeMsgCreateOracleMarket(marketId, pair, exponent, minExchanges, minPriceChangePpm, exchangeConfigJson) {
        const msg = {
            // uses x/gov module account since creating the oracle market is a governance action.
            authority: constants_1.GOV_MODULE_ADDRESS,
            params: {
                id: marketId,
                pair,
                exponent,
                minExchanges,
                minPriceChangePpm,
                exchangeConfigJson,
            },
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_CREATE_ORACLE_MARKET,
            value: msg,
        };
    }
    // ------------ x/perpetuals ------------
    composeMsgCreatePerpetual(perpetualId, marketId, ticker, atomicResolution, liquidityTier, marketType) {
        const msg = {
            // uses x/gov module account since creating the perpetual is a governance action.
            authority: constants_1.GOV_MODULE_ADDRESS,
            params: {
                id: perpetualId,
                marketId,
                ticker,
                atomicResolution,
                defaultFundingPpm: 0,
                liquidityTier,
                marketType,
            },
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_CREATE_PERPETUAL,
            value: msg,
        };
    }
    // ------------ x/delaymsg ------------
    composeMsgDelayMessage(embeddedMsg, delayBlocks) {
        const msg = {
            // all msgs sent to x/delay must be from x/gov module account.
            authority: constants_1.GOV_MODULE_ADDRESS,
            msg: embeddedMsg,
            delayBlocks,
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_DELAY_MESSAGE,
            value: msg,
        };
    }
    // ------------ x/gov ------------
    composeMsgSubmitProposal(title, initialDepositAmount, initialDepositDenomConfig, summary, messages, proposer, metadata = '', expedited = false) {
        const initialDeposit = [
            {
                amount: initialDepositAmount,
                denom: initialDepositDenomConfig.CHAINTOKEN_DENOM,
            },
        ];
        const msg = {
            title,
            initialDeposit,
            summary,
            messages,
            proposer,
            metadata,
            expedited,
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_SUBMIT_PROPOSAL,
            value: msg,
        };
    }
    // ------------ x/vault ------------
    composeMsgDepositToMegavault(address, subaccountNumber, quoteQuantums) {
        const subaccountId = {
            owner: address,
            number: subaccountNumber,
        };
        const msg = {
            quoteQuantums,
            subaccountId,
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_DEPOSIT_TO_MEGAVAULT,
            value: msg,
        };
    }
    composeMsgWithdrawFromMegavault(address, subaccountNumber, shares, minQuoteQuantums) {
        const subaccountId = {
            owner: address,
            number: subaccountNumber,
        };
        const msg = {
            minQuoteQuantums,
            shares: { numShares: shares },
            subaccountId,
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_WITHDRAW_FROM_MEGAVAULT,
            value: msg,
        };
    }
    // ------------ x/staking ------------
    composeMsgDelegate(delegator, validator, amount) {
        const msg = {
            delegatorAddress: delegator,
            validatorAddress: validator,
            amount,
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_DELEGATE,
            value: msg,
        };
    }
    composeMsgUndelegate(delegator, validator, amount) {
        const msg = {
            delegatorAddress: delegator,
            validatorAddress: validator,
            amount,
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_UNDELEGATE,
            value: msg,
        };
    }
    // ------------ x/distribution ------------
    composeMsgWithdrawDelegatorReward(delegator, validator) {
        const msg = {
            delegatorAddress: delegator,
            validatorAddress: validator,
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_WITHDRAW_DELEGATOR_REWARD,
            value: msg,
        };
    }
    // ------------ x/affiliates ------------
    composeMsgRegisterAffiliate(referee, affiliate) {
        const msg = {
            referee,
            affiliate,
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_REGISTER_AFFILIATE,
            value: msg,
        };
    }
    // ------------ x/listing ------------
    composeMsgCreateMarketPermissionless(address, ticker, subaccountNumber) {
        const subaccountId = {
            owner: address,
            number: subaccountNumber,
        };
        const msg = {
            ticker,
            subaccountId
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_CREATE_MARKET_PERMISSIONLESS,
            value: msg,
        };
    }
    // ----------- x/accountplus --------
    composeMsgAddAuthenticator(address, authenticatorType, data) {
        const msg = {
            sender: address,
            authenticatorType,
            data,
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_ADD_AUTHENTICATOR,
            value: msg,
        };
    }
    composeMsgRemoveAuthenticator(address, id) {
        const msg = {
            sender: address,
            id,
        };
        return {
            typeUrl: constants_1.TYPE_URL_MSG_REMOVE_AUTHENTICATOR,
            value: msg,
        };
    }
    // ------------ util ------------
    validateGoodTilBlockAndTime(orderFlags, goodTilBlock, goodTilBlockTime) {
        if (orderFlags === 0 && goodTilBlock === 0) {
            throw new Error('goodTilBlock must be set if orderFlags is 0');
        }
        else if (orderFlags !== 0 && goodTilBlockTime === 0) {
            throw new Error('goodTilBlockTime must be set if orderFlags is not 0');
        }
    }
    wrapMessageAsAny(registry, message) {
        return registry.encodeAsAny(message);
    }
    wrapMessageArrAsAny(registry, messages) {
        const encodedMessages = messages.map((message) => this.wrapMessageAsAny(registry, message));
        return encodedMessages;
    }
}
exports.Composer = Composer;
