// @dydxprotocol/v4-client-js/build/src/clients/modules/local-wallet.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const amino_1 = require("@cosmjs/amino");
const proto_signing_1 = require("@cosmjs/proto-signing");
const stargate_1 = require("@cosmjs/stargate");
const long_1 = __importDefault(require("long"));
const protobufjs_1 = __importDefault(require("protobufjs"));
const registry_1 = require("../lib/registry");
const signer_1 = require("./signer");
// Required for encoding and decoding queries that are of type Long.
protobufjs_1.default.util.Long = long_1.default;
protobufjs_1.default.configure();
class LocalWallet {
    static async fromOfflineSigner(signer) {
        const wallet = new LocalWallet();
        await wallet.setSigner(signer);
        return wallet;
    }
    static async fromMnemonic(mnemonic, prefix) {
        const wallet = new LocalWallet();
        await wallet.setMnemonic(mnemonic, prefix);
        return wallet;
    }
    async setSigner(signer) {
        this.offlineSigner = signer;
        const stargateClient = await stargate_1.SigningStargateClient.offline(signer, {
            registry: (0, registry_1.generateRegistry)(),
        });
        const accountData = await signer.getAccounts();
        const firstAccount = accountData[0];
        this.accounts = [...accountData];
        this.address = firstAccount.address;
        this.pubKey = (0, amino_1.encodeSecp256k1Pubkey)(firstAccount.pubkey);
        this.signer = new signer_1.TransactionSigner(this.address, stargateClient, signer);
    }
    async setMnemonic(mnemonic, prefix) {
        const signer = await proto_signing_1.DirectSecp256k1HdWallet.fromMnemonic(mnemonic, { prefix });
        return this.setSigner(signer);
    }
    async signTransaction(messages, transactionOptions, fee, memo = '') {
        return this.signer.signTransaction(messages, transactionOptions, fee, memo, this.pubKey);
    }
}
exports.default = LocalWallet;
