// @dydxprotocol/v4-client-js/build/src/lib/utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.calculateClockOffsetFromFetchDateHeader = exports.getGovAddNewMarketSummary = exports.getGovAddNewMarketTitle = exports.sleep = exports.clientIdFromString = exports.generateRandomClientId = exports.randomInt = void 0;
const constants_1 = require("../clients/constants");
/**
 * Returns a random integer value between 0 and (n-1).
 */
function randomInt(n) {
    return Math.floor(Math.random() * n);
}
exports.randomInt = randomInt;
/**
 * Generate a random clientId.
 */
function generateRandomClientId() {
    return randomInt(constants_1.MAX_UINT_32 + 1);
}
exports.generateRandomClientId = generateRandomClientId;
/**
 * Deterministically generate a valid clientId from an arbitrary string by performing a
 * quick hashing function on the string.
 */
function clientIdFromString(input) {
    let hash = 0;
    if (input.length === 0)
        return hash;
    for (let i = 0; i < input.length; i++) {
        hash = (hash << 5) - hash + input.charCodeAt(i); // eslint-disable-line no-bitwise
        hash |= 0; // eslint-disable-line no-bitwise
    }
    // Bitwise operators covert the value to a 32-bit integer.
    // We must coerce this into a 32-bit unsigned integer.
    return hash + 2 ** 31;
}
exports.clientIdFromString = clientIdFromString;
/**
 * Pauses the execution of the program for a specified time.
 * @param ms - The number of milliseconds to pause the program.
 * @returns A promise that resolves after the specified number of milliseconds.
 */
async function sleep(ms) {
    return new Promise((resolve) => setTimeout(resolve, ms));
}
exports.sleep = sleep;
/**
 * Returns a title to use for a gov proposal that adds a new market.
 *
 * @param ticker ticker symbol for the new market.
 * @returns title for the gov proposal.
 */
function getGovAddNewMarketTitle(ticker) {
    return `Add ${ticker} perpetual market`;
}
exports.getGovAddNewMarketTitle = getGovAddNewMarketTitle;
/**
 * Returns a summary to use for a gov proposal that adds a new market.
 *
 * @param ticker ticker symbol for the new market.
 * @param delayBlocks number of blocks to wait before activating the market.
 * @returns summary for the gov proposal.
 */
function getGovAddNewMarketSummary(ticker, delayBlocks) {
    return `Add the x/prices, x/perpetuals and x/clob parameters needed for a ${ticker} perpetual market. Create the market in INITIALIZING status and transition it to ACTIVE status after ${delayBlocks} blocks.`;
}
exports.getGovAddNewMarketSummary = getGovAddNewMarketSummary;
function calculateClockOffsetFromFetchDateHeader(clientRequestStartTime, serverReportedTime, clientRequestEndTime) {
    // use midpoint, so assume that time for request reach server === time for response to reach us
    const estimatedLocalTimeAtServerArrival = (clientRequestStartTime + clientRequestEndTime) / 2;
    // we need an offset such that estimatedLocalTimeAtServerArrival + offset = serverReportedTime
    const offset = serverReportedTime - estimatedLocalTimeAtServerArrival;
    return Math.floor(offset);
}
exports.calculateClockOffsetFromFetchDateHeader = calculateClockOffsetFromFetchDateHeader;
