// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/rpc.tx.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createRPCMsgClient = void 0;
const createRPCMsgClient = async ({ rpc }) => ({
    cosmos: {
        auth: {
            v1beta1: new (await Promise.resolve().then(() => __importStar(require("../cosmos/auth/v1beta1/tx.rpc.msg")))).MsgClientImpl(rpc)
        },
        authz: {
            v1beta1: new (await Promise.resolve().then(() => __importStar(require("../cosmos/authz/v1beta1/tx.rpc.msg")))).MsgClientImpl(rpc)
        },
        bank: {
            v1beta1: new (await Promise.resolve().then(() => __importStar(require("../cosmos/bank/v1beta1/tx.rpc.msg")))).MsgClientImpl(rpc)
        },
        circuit: {
            v1: new (await Promise.resolve().then(() => __importStar(require("../cosmos/circuit/v1/tx.rpc.msg")))).MsgClientImpl(rpc)
        },
        consensus: {
            v1: new (await Promise.resolve().then(() => __importStar(require("../cosmos/consensus/v1/tx.rpc.msg")))).MsgClientImpl(rpc)
        },
        crisis: {
            v1beta1: new (await Promise.resolve().then(() => __importStar(require("../cosmos/crisis/v1beta1/tx.rpc.msg")))).MsgClientImpl(rpc)
        },
        distribution: {
            v1beta1: new (await Promise.resolve().then(() => __importStar(require("../cosmos/distribution/v1beta1/tx.rpc.msg")))).MsgClientImpl(rpc)
        },
        evidence: {
            v1beta1: new (await Promise.resolve().then(() => __importStar(require("../cosmos/evidence/v1beta1/tx.rpc.msg")))).MsgClientImpl(rpc)
        },
        feegrant: {
            v1beta1: new (await Promise.resolve().then(() => __importStar(require("../cosmos/feegrant/v1beta1/tx.rpc.msg")))).MsgClientImpl(rpc)
        },
        gov: {
            v1: new (await Promise.resolve().then(() => __importStar(require("../cosmos/gov/v1/tx.rpc.msg")))).MsgClientImpl(rpc),
            v1beta1: new (await Promise.resolve().then(() => __importStar(require("../cosmos/gov/v1beta1/tx.rpc.msg")))).MsgClientImpl(rpc)
        },
        group: {
            v1: new (await Promise.resolve().then(() => __importStar(require("../cosmos/group/v1/tx.rpc.msg")))).MsgClientImpl(rpc)
        },
        mint: {
            v1beta1: new (await Promise.resolve().then(() => __importStar(require("../cosmos/mint/v1beta1/tx.rpc.msg")))).MsgClientImpl(rpc)
        },
        nft: {
            v1beta1: new (await Promise.resolve().then(() => __importStar(require("../cosmos/nft/v1beta1/tx.rpc.msg")))).MsgClientImpl(rpc)
        },
        slashing: {
            v1beta1: new (await Promise.resolve().then(() => __importStar(require("../cosmos/slashing/v1beta1/tx.rpc.msg")))).MsgClientImpl(rpc)
        },
        staking: {
            v1beta1: new (await Promise.resolve().then(() => __importStar(require("../cosmos/staking/v1beta1/tx.rpc.msg")))).MsgClientImpl(rpc)
        },
        upgrade: {
            v1beta1: new (await Promise.resolve().then(() => __importStar(require("../cosmos/upgrade/v1beta1/tx.rpc.msg")))).MsgClientImpl(rpc)
        },
        vesting: {
            v1beta1: new (await Promise.resolve().then(() => __importStar(require("../cosmos/vesting/v1beta1/tx.rpc.msg")))).MsgClientImpl(rpc)
        }
    },
    dydxprotocol: {
        accountplus: new (await Promise.resolve().then(() => __importStar(require("./accountplus/tx.rpc.msg")))).MsgClientImpl(rpc),
        affiliates: new (await Promise.resolve().then(() => __importStar(require("./affiliates/tx.rpc.msg")))).MsgClientImpl(rpc),
        blocktime: new (await Promise.resolve().then(() => __importStar(require("./blocktime/tx.rpc.msg")))).MsgClientImpl(rpc),
        bridge: new (await Promise.resolve().then(() => __importStar(require("./bridge/tx.rpc.msg")))).MsgClientImpl(rpc),
        clob: new (await Promise.resolve().then(() => __importStar(require("./clob/tx.rpc.msg")))).MsgClientImpl(rpc),
        delaymsg: new (await Promise.resolve().then(() => __importStar(require("./delaymsg/tx.rpc.msg")))).MsgClientImpl(rpc),
        feetiers: new (await Promise.resolve().then(() => __importStar(require("./feetiers/tx.rpc.msg")))).MsgClientImpl(rpc),
        govplus: new (await Promise.resolve().then(() => __importStar(require("./govplus/tx.rpc.msg")))).MsgClientImpl(rpc),
        listing: new (await Promise.resolve().then(() => __importStar(require("./listing/tx.rpc.msg")))).MsgClientImpl(rpc),
        perpetuals: new (await Promise.resolve().then(() => __importStar(require("./perpetuals/tx.rpc.msg")))).MsgClientImpl(rpc),
        prices: new (await Promise.resolve().then(() => __importStar(require("./prices/tx.rpc.msg")))).MsgClientImpl(rpc),
        ratelimit: new (await Promise.resolve().then(() => __importStar(require("./ratelimit/tx.rpc.msg")))).MsgClientImpl(rpc),
        revshare: new (await Promise.resolve().then(() => __importStar(require("./revshare/tx.rpc.msg")))).MsgClientImpl(rpc),
        rewards: new (await Promise.resolve().then(() => __importStar(require("./rewards/tx.rpc.msg")))).MsgClientImpl(rpc),
        sending: new (await Promise.resolve().then(() => __importStar(require("./sending/tx.rpc.msg")))).MsgClientImpl(rpc),
        stats: new (await Promise.resolve().then(() => __importStar(require("./stats/tx.rpc.msg")))).MsgClientImpl(rpc),
        vault: new (await Promise.resolve().then(() => __importStar(require("./vault/tx.rpc.msg")))).MsgClientImpl(rpc),
        vest: new (await Promise.resolve().then(() => __importStar(require("./vest/tx.rpc.msg")))).MsgClientImpl(rpc)
    }
});
exports.createRPCMsgClient = createRPCMsgClient;
