// @dydxprotocol/v4-client-js/build/examples/batch_cancel_orders_example.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const lodash_1 = __importDefault(require("lodash"));
const src_1 = require("../src");
const composite_client_1 = require("../src/clients/composite-client");
const constants_1 = require("../src/clients/constants");
const local_wallet_1 = __importDefault(require("../src/clients/modules/local-wallet"));
const subaccount_1 = require("../src/clients/subaccount");
const utils_1 = require("../src/lib/utils");
const constants_2 = require("./constants");
const generateShortTermOrdersInfo = () => [
    {
        marketId: 'ETH-USD',
        clientId: (0, utils_1.randomInt)(constants_2.MAX_CLIENT_ID),
        side: constants_1.OrderSide.SELL,
        price: 4000,
        size: 0.01,
    },
    {
        marketId: 'ETH-USD',
        clientId: (0, utils_1.randomInt)(constants_2.MAX_CLIENT_ID),
        side: constants_1.OrderSide.SELL,
        price: 4200,
        size: 0.02,
    },
    {
        marketId: 'BTC-USD',
        clientId: (0, utils_1.randomInt)(constants_2.MAX_CLIENT_ID),
        side: constants_1.OrderSide.BUY,
        price: 40000,
        size: 0.01,
    },
];
const generateBatchCancelShortTermOrders = (ordersInfo) => {
    const ordersGroupedByMarketIds = lodash_1.default.groupBy(ordersInfo, (info) => info.marketId);
    return Object.keys(ordersGroupedByMarketIds).map((marketId) => ({
        marketId,
        clientIds: ordersGroupedByMarketIds[marketId].map((info) => info.clientId),
    }));
};
async function test() {
    try {
        const wallet = await local_wallet_1.default.fromMnemonic(constants_2.DYDX_TEST_MNEMONIC, src_1.BECH32_PREFIX);
        console.log('**Wallet**', wallet);
        const network = constants_1.Network.testnet();
        const client = await composite_client_1.CompositeClient.connect(network);
        console.log('**Client**', client);
        const subaccount = new subaccount_1.SubaccountInfo(wallet, 0);
        const currentBlock = await client.validatorClient.get.latestBlockHeight();
        const goodTilBlock = currentBlock + 10;
        const shortTermOrdersInfo = generateShortTermOrdersInfo();
        await placeShortTermOrders(client, subaccount, shortTermOrdersInfo, goodTilBlock);
        await (0, utils_1.sleep)(5000);
        await batchCancelOrders(client, subaccount, shortTermOrdersInfo, goodTilBlock);
    }
    catch (error) {
        console.error('**Test Failed**', error.message);
    }
}
const placeShortTermOrders = async (client, subaccount, shortTermOrdersInfo, goodTilBlock) => {
    const orderPromises = shortTermOrdersInfo.map(async (order) => {
        try {
            const tx = await client.placeShortTermOrder(subaccount, order.marketId, order.side, order.price, order.size, order.clientId, goodTilBlock, src_1.Order_TimeInForce.TIME_IN_FORCE_UNSPECIFIED, false);
            console.log('**Short Term Order Tx**', tx.hash);
        }
        catch (error) {
            console.error(`**Short Term Order Failed for Market ${order.marketId}, Client ID ${order.clientId}**`, error.message);
        }
    });
    // Wait for all order placements to complete
    await Promise.all(orderPromises);
};
const batchCancelOrders = async (client, subaccount, shortTermOrdersInfo, goodTilBlock) => {
    const shortTermOrdersPayload = generateBatchCancelShortTermOrders(shortTermOrdersInfo);
    try {
        const tx = await client.batchCancelShortTermOrdersWithMarketId(subaccount, shortTermOrdersPayload, goodTilBlock + 10);
        console.log('**Batch Cancel Short Term Orders Tx**', tx);
    }
    catch (error) {
        console.error('**Batch Cancel Short Term Orders Failed**', error.message);
    }
};
test()
    .then(() => {
    console.log('**Batch Cancel Test Completed Successfully**');
})
    .catch((error) => {
    console.error('**Batch Cancel Test Execution Error**', error.message);
});
