// @dydxprotocol/v4-client-js/build/examples/noble_example.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const long_1 = __importDefault(require("long"));
const constants_1 = require("../src/clients/constants");
const local_wallet_1 = __importDefault(require("../src/clients/modules/local-wallet"));
const noble_client_1 = require("../src/clients/noble-client");
const validator_client_1 = require("../src/clients/validator-client");
const constants_2 = require("../src/lib/constants");
const utils_1 = require("../src/lib/utils");
const constants_3 = require("./constants");
async function test() {
    const dydxClient = await validator_client_1.ValidatorClient.connect(constants_1.Network.testnet().validatorConfig);
    const dydxWallet = await local_wallet_1.default.fromMnemonic(constants_3.DYDX_TEST_MNEMONIC, constants_2.BECH32_PREFIX);
    const nobleWallet = await local_wallet_1.default.fromMnemonic(constants_3.DYDX_TEST_MNEMONIC, constants_2.NOBLE_BECH32_PREFIX);
    const client = new noble_client_1.NobleClient('https://rpc.testnet.noble.strange.love', 'Noble example');
    await client.connect(nobleWallet);
    if (nobleWallet.address === undefined || dydxWallet.address === undefined) {
        throw new Error('Wallet not found');
    }
    // IBC to noble
    const ibcToNobleMsg = {
        typeUrl: '/ibc.applications.transfer.v1.MsgTransfer',
        value: {
            sourcePort: 'transfer',
            sourceChannel: 'channel-0',
            token: {
                denom: 'ibc/8E27BA2D5493AF5636760E354E46004562C46AB7EC0CC4C1CA14E9E20E2545B5',
                amount: '1000000',
            },
            sender: dydxWallet.address,
            receiver: nobleWallet.address,
            timeoutTimestamp: long_1.default.fromNumber(Math.floor(Date.now() / 1000) * 1e9 + 10 * 60 * 1e9),
        },
    };
    const msgs = [ibcToNobleMsg];
    const encodeObjects = new Promise((resolve) => resolve(msgs));
    await dydxClient.post.send(dydxWallet, () => {
        return encodeObjects;
    }, false, undefined, undefined);
    await (0, utils_1.sleep)(30000);
    try {
        const coins = await client.getAccountBalances();
        console.log('Balances');
        console.log(JSON.stringify(coins));
        // IBC from noble
        const ibcFromNobleMsg = {
            typeUrl: '/ibc.applications.transfer.v1.MsgTransfer',
            value: {
                sourcePort: 'transfer',
                sourceChannel: 'channel-21',
                token: {
                    denom: 'uusdc',
                    amount: coins[0].amount,
                },
                sender: nobleWallet.address,
                receiver: dydxWallet.address,
                timeoutTimestamp: long_1.default.fromNumber(Math.floor(Date.now() / 1000) * 1e9 + 10 * 60 * 1e9),
            },
        };
        const fee = await client.simulateTransaction([ibcFromNobleMsg]);
        ibcFromNobleMsg.value.token.amount = (parseInt(ibcFromNobleMsg.value.token.amount, 10) -
            Math.floor(parseInt(fee.amount[0].amount, 10) * 1.4)).toString();
        await client.send([ibcFromNobleMsg]);
    }
    catch (error) {
        console.log(JSON.stringify(error.message));
    }
    await (0, utils_1.sleep)(30000);
    try {
        const coin = await client.getAccountBalance('uusdc');
        console.log('Balance');
        console.log(JSON.stringify(coin));
    }
    catch (error) {
        console.log(JSON.stringify(error.message));
    }
}
test()
    .then(() => { })
    .catch((error) => {
    console.log(error.message);
    console.log(error);
});
