// @dydxprotocol/v4-client-js/build/examples/permissioned_keys_example.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const util_1 = require("util");
const encoding_1 = require("@cosmjs/encoding");
const order_1 = require("@dydxprotocol/v4-proto/src/codegen/dydxprotocol/clob/order");
const src_1 = require("../src");
const composite_client_1 = require("../src/clients/composite-client");
const constants_1 = require("../src/clients/constants");
const local_wallet_1 = __importDefault(require("../src/clients/modules/local-wallet"));
const subaccount_1 = require("../src/clients/subaccount");
const constants_2 = require("./constants");
async function test() {
    const wallet1 = await local_wallet_1.default.fromMnemonic(constants_2.DYDX_TEST_MNEMONIC, src_1.BECH32_PREFIX);
    const wallet2 = await local_wallet_1.default.fromMnemonic(constants_2.DYDX_TEST_MNEMONIC_2, src_1.BECH32_PREFIX);
    const network = constants_1.Network.staging();
    const client = await composite_client_1.CompositeClient.connect(network);
    client.setSelectedGasDenom(constants_1.SelectedGasDenom.NATIVE);
    console.log('**Client**');
    console.log(client);
    const subaccount1 = new subaccount_1.SubaccountInfo(wallet1, 0);
    const subaccount2 = new subaccount_1.SubaccountInfo(wallet2, 0);
    // Change second wallet pubkey
    // Add an authenticator to allow wallet2 to place orders
    console.log("** Adding authenticator **");
    await addAuthenticator(client, subaccount1, wallet2.pubKey.value);
    const authenticators = await client.getAuthenticators(wallet1.address);
    // Last element in authenticators array is the most recently created
    const lastElement = authenticators.accountAuthenticators.length - 1;
    const authenticatorID = authenticators.accountAuthenticators[lastElement].id;
    // Placing order using subaccount2 for subaccount1 succeeds
    console.log("** Placing order with authenticator **");
    await placeOrder(client, subaccount2, subaccount1, authenticatorID);
    // Remove authenticator
    console.log("** Removing authenticator **");
    await removeAuthenticator(client, subaccount1, authenticatorID);
    // Placing an order using subaccount2 will now fail
    console.log("** Placing order with invalid authenticator should fail **");
    await placeOrder(client, subaccount2, subaccount1, authenticatorID);
}
async function removeAuthenticator(client, subaccount, id) {
    await client.removeAuthenticator(subaccount, id);
}
async function addAuthenticator(client, subaccount, authedPubKey) {
    const subAuth = [{
            type: constants_1.AuthenticatorType.SIGNATURE_VERIFICATION,
            config: authedPubKey,
        },
        {
            type: constants_1.AuthenticatorType.ANY_OF,
            config: [
                {
                    type: constants_1.AuthenticatorType.MESSAGE_FILTER,
                    config: (0, encoding_1.toBase64)(new util_1.TextEncoder().encode('/dydxprotocol.clob.MsgPlaceOrder')),
                },
                {
                    type: constants_1.AuthenticatorType.MESSAGE_FILTER,
                    config: (0, encoding_1.toBase64)(new util_1.TextEncoder().encode('/dydxprotocol.clob.MsgPlaceOrder')),
                },
            ]
        }
    ];
    const jsonString = JSON.stringify(subAuth);
    const encodedData = new util_1.TextEncoder().encode(jsonString);
    try {
        await client.addAuthenticator(subaccount, constants_1.AuthenticatorType.ALL_OF, encodedData);
    }
    catch (error) {
        console.log(error.message);
    }
}
async function placeOrder(client, fromAccount, forAccount, authenticatorId) {
    try {
        const side = constants_1.OrderSide.BUY;
        const price = Number('1000');
        const currentBlock = await client.validatorClient.get.latestBlockHeight();
        const nextValidBlockHeight = currentBlock + 5;
        const goodTilBlock = nextValidBlockHeight + 10;
        const timeInForce = order_1.Order_TimeInForce.TIME_IN_FORCE_UNSPECIFIED;
        const clientId = Math.floor(Math.random() * 10000);
        const tx = await client.placeShortTermOrder(fromAccount, 'ETH-USD', side, price, 0.01, clientId, goodTilBlock, timeInForce, false, undefined, {
            authenticators: [authenticatorId],
            accountForOrder: forAccount,
        });
        console.log('**Order Tx**');
        console.log(Buffer.from(tx.hash).toString('hex'));
    }
    catch (error) {
        console.log(error.message);
    }
}
test()
    .then(() => { })
    .catch((error) => {
    console.log(error.message);
});
