// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/vault/genesis.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VaultV6 = exports.GenesisStateV6 = exports.Vault = exports.GenesisState = void 0;
const share_1 = require("./share");
const params_1 = require("./params");
const vault_1 = require("./vault");
const _m0 = __importStar(require("protobufjs/minimal"));
function createBaseGenesisState() {
    return {
        totalShares: undefined,
        ownerShares: [],
        vaults: [],
        defaultQuotingParams: undefined,
        allOwnerShareUnlocks: [],
        operatorParams: undefined
    };
}
exports.GenesisState = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.totalShares !== undefined) {
            share_1.NumShares.encode(message.totalShares, writer.uint32(10).fork()).ldelim();
        }
        for (const v of message.ownerShares) {
            share_1.OwnerShare.encode(v, writer.uint32(18).fork()).ldelim();
        }
        for (const v of message.vaults) {
            exports.Vault.encode(v, writer.uint32(26).fork()).ldelim();
        }
        if (message.defaultQuotingParams !== undefined) {
            params_1.QuotingParams.encode(message.defaultQuotingParams, writer.uint32(34).fork()).ldelim();
        }
        for (const v of message.allOwnerShareUnlocks) {
            share_1.OwnerShareUnlocks.encode(v, writer.uint32(42).fork()).ldelim();
        }
        if (message.operatorParams !== undefined) {
            params_1.OperatorParams.encode(message.operatorParams, writer.uint32(50).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGenesisState();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.totalShares = share_1.NumShares.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.ownerShares.push(share_1.OwnerShare.decode(reader, reader.uint32()));
                    break;
                case 3:
                    message.vaults.push(exports.Vault.decode(reader, reader.uint32()));
                    break;
                case 4:
                    message.defaultQuotingParams = params_1.QuotingParams.decode(reader, reader.uint32());
                    break;
                case 5:
                    message.allOwnerShareUnlocks.push(share_1.OwnerShareUnlocks.decode(reader, reader.uint32()));
                    break;
                case 6:
                    message.operatorParams = params_1.OperatorParams.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseGenesisState();
        message.totalShares = object.totalShares !== undefined && object.totalShares !== null ? share_1.NumShares.fromPartial(object.totalShares) : undefined;
        message.ownerShares = ((_a = object.ownerShares) === null || _a === void 0 ? void 0 : _a.map(e => share_1.OwnerShare.fromPartial(e))) || [];
        message.vaults = ((_b = object.vaults) === null || _b === void 0 ? void 0 : _b.map(e => exports.Vault.fromPartial(e))) || [];
        message.defaultQuotingParams = object.defaultQuotingParams !== undefined && object.defaultQuotingParams !== null ? params_1.QuotingParams.fromPartial(object.defaultQuotingParams) : undefined;
        message.allOwnerShareUnlocks = ((_c = object.allOwnerShareUnlocks) === null || _c === void 0 ? void 0 : _c.map(e => share_1.OwnerShareUnlocks.fromPartial(e))) || [];
        message.operatorParams = object.operatorParams !== undefined && object.operatorParams !== null ? params_1.OperatorParams.fromPartial(object.operatorParams) : undefined;
        return message;
    }
};
function createBaseVault() {
    return {
        vaultId: undefined,
        vaultParams: undefined,
        mostRecentClientIds: []
    };
}
exports.Vault = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.vaultId !== undefined) {
            vault_1.VaultId.encode(message.vaultId, writer.uint32(10).fork()).ldelim();
        }
        if (message.vaultParams !== undefined) {
            params_1.VaultParams.encode(message.vaultParams, writer.uint32(18).fork()).ldelim();
        }
        writer.uint32(26).fork();
        for (const v of message.mostRecentClientIds) {
            writer.uint32(v);
        }
        writer.ldelim();
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseVault();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.vaultId = vault_1.VaultId.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.vaultParams = params_1.VaultParams.decode(reader, reader.uint32());
                    break;
                case 3:
                    if ((tag & 7) === 2) {
                        const end2 = reader.uint32() + reader.pos;
                        while (reader.pos < end2) {
                            message.mostRecentClientIds.push(reader.uint32());
                        }
                    }
                    else {
                        message.mostRecentClientIds.push(reader.uint32());
                    }
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseVault();
        message.vaultId = object.vaultId !== undefined && object.vaultId !== null ? vault_1.VaultId.fromPartial(object.vaultId) : undefined;
        message.vaultParams = object.vaultParams !== undefined && object.vaultParams !== null ? params_1.VaultParams.fromPartial(object.vaultParams) : undefined;
        message.mostRecentClientIds = ((_a = object.mostRecentClientIds) === null || _a === void 0 ? void 0 : _a.map(e => e)) || [];
        return message;
    }
};
function createBaseGenesisStateV6() {
    return {
        vaults: [],
        defaultQuotingParams: undefined
    };
}
exports.GenesisStateV6 = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.vaults) {
            exports.Vault.encode(v, writer.uint32(18).fork()).ldelim();
        }
        if (message.defaultQuotingParams !== undefined) {
            params_1.QuotingParams.encode(message.defaultQuotingParams, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGenesisStateV6();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 2:
                    message.vaults.push(exports.Vault.decode(reader, reader.uint32()));
                    break;
                case 3:
                    message.defaultQuotingParams = params_1.QuotingParams.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseGenesisStateV6();
        message.vaults = ((_a = object.vaults) === null || _a === void 0 ? void 0 : _a.map(e => exports.Vault.fromPartial(e))) || [];
        message.defaultQuotingParams = object.defaultQuotingParams !== undefined && object.defaultQuotingParams !== null ? params_1.QuotingParams.fromPartial(object.defaultQuotingParams) : undefined;
        return message;
    }
};
function createBaseVaultV6() {
    return {
        vaultId: undefined,
        totalShares: undefined,
        ownerShares: [],
        vaultParams: undefined,
        mostRecentClientIds: []
    };
}
exports.VaultV6 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.vaultId !== undefined) {
            vault_1.VaultId.encode(message.vaultId, writer.uint32(10).fork()).ldelim();
        }
        if (message.totalShares !== undefined) {
            share_1.NumShares.encode(message.totalShares, writer.uint32(18).fork()).ldelim();
        }
        for (const v of message.ownerShares) {
            share_1.OwnerShare.encode(v, writer.uint32(26).fork()).ldelim();
        }
        if (message.vaultParams !== undefined) {
            params_1.VaultParams.encode(message.vaultParams, writer.uint32(34).fork()).ldelim();
        }
        writer.uint32(42).fork();
        for (const v of message.mostRecentClientIds) {
            writer.uint32(v);
        }
        writer.ldelim();
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseVaultV6();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.vaultId = vault_1.VaultId.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.totalShares = share_1.NumShares.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.ownerShares.push(share_1.OwnerShare.decode(reader, reader.uint32()));
                    break;
                case 4:
                    message.vaultParams = params_1.VaultParams.decode(reader, reader.uint32());
                    break;
                case 5:
                    if ((tag & 7) === 2) {
                        const end2 = reader.uint32() + reader.pos;
                        while (reader.pos < end2) {
                            message.mostRecentClientIds.push(reader.uint32());
                        }
                    }
                    else {
                        message.mostRecentClientIds.push(reader.uint32());
                    }
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseVaultV6();
        message.vaultId = object.vaultId !== undefined && object.vaultId !== null ? vault_1.VaultId.fromPartial(object.vaultId) : undefined;
        message.totalShares = object.totalShares !== undefined && object.totalShares !== null ? share_1.NumShares.fromPartial(object.totalShares) : undefined;
        message.ownerShares = ((_a = object.ownerShares) === null || _a === void 0 ? void 0 : _a.map(e => share_1.OwnerShare.fromPartial(e))) || [];
        message.vaultParams = object.vaultParams !== undefined && object.vaultParams !== null ? params_1.VaultParams.fromPartial(object.vaultParams) : undefined;
        message.mostRecentClientIds = ((_b = object.mostRecentClientIds) === null || _b === void 0 ? void 0 : _b.map(e => e)) || [];
        return message;
    }
};
