// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/vault/share.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShareUnlock = exports.OwnerShareUnlocks = exports.OwnerShare = exports.NumShares = void 0;
const _m0 = __importStar(require("protobufjs/minimal"));
function createBaseNumShares() {
    return {
        numShares: new Uint8Array()
    };
}
exports.NumShares = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.numShares.length !== 0) {
            writer.uint32(18).bytes(message.numShares);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseNumShares();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 2:
                    message.numShares = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseNumShares();
        message.numShares = (_a = object.numShares) !== null && _a !== void 0 ? _a : new Uint8Array();
        return message;
    }
};
function createBaseOwnerShare() {
    return {
        owner: "",
        shares: undefined
    };
}
exports.OwnerShare = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.owner !== "") {
            writer.uint32(10).string(message.owner);
        }
        if (message.shares !== undefined) {
            exports.NumShares.encode(message.shares, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseOwnerShare();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.owner = reader.string();
                    break;
                case 2:
                    message.shares = exports.NumShares.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseOwnerShare();
        message.owner = (_a = object.owner) !== null && _a !== void 0 ? _a : "";
        message.shares = object.shares !== undefined && object.shares !== null ? exports.NumShares.fromPartial(object.shares) : undefined;
        return message;
    }
};
function createBaseOwnerShareUnlocks() {
    return {
        ownerAddress: "",
        shareUnlocks: []
    };
}
exports.OwnerShareUnlocks = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.ownerAddress !== "") {
            writer.uint32(10).string(message.ownerAddress);
        }
        for (const v of message.shareUnlocks) {
            exports.ShareUnlock.encode(v, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseOwnerShareUnlocks();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.ownerAddress = reader.string();
                    break;
                case 2:
                    message.shareUnlocks.push(exports.ShareUnlock.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseOwnerShareUnlocks();
        message.ownerAddress = (_a = object.ownerAddress) !== null && _a !== void 0 ? _a : "";
        message.shareUnlocks = ((_b = object.shareUnlocks) === null || _b === void 0 ? void 0 : _b.map(e => exports.ShareUnlock.fromPartial(e))) || [];
        return message;
    }
};
function createBaseShareUnlock() {
    return {
        shares: undefined,
        unlockBlockHeight: 0
    };
}
exports.ShareUnlock = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.shares !== undefined) {
            exports.NumShares.encode(message.shares, writer.uint32(10).fork()).ldelim();
        }
        if (message.unlockBlockHeight !== 0) {
            writer.uint32(16).uint32(message.unlockBlockHeight);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseShareUnlock();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.shares = exports.NumShares.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.unlockBlockHeight = reader.uint32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseShareUnlock();
        message.shares = object.shares !== undefined && object.shares !== null ? exports.NumShares.fromPartial(object.shares) : undefined;
        message.unlockBlockHeight = (_a = object.unlockBlockHeight) !== null && _a !== void 0 ? _a : 0;
        return message;
    }
};
