// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/vault/tx.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgSetVaultQuotingParams = exports.MsgUpdateParams = exports.MsgRetrieveFromVaultResponse = exports.MsgRetrieveFromVault = exports.MsgAllocateToVaultResponse = exports.MsgAllocateToVault = exports.MsgUpdateOperatorParamsResponse = exports.MsgUpdateOperatorParams = exports.MsgUnlockSharesResponse = exports.MsgUnlockShares = exports.MsgSetVaultParamsResponse = exports.MsgSetVaultParams = exports.MsgUpdateDefaultQuotingParamsResponse = exports.MsgUpdateDefaultQuotingParams = exports.MsgWithdrawFromMegavaultResponse = exports.MsgWithdrawFromMegavault = exports.MsgDepositToMegavaultResponse = exports.MsgDepositToMegavault = void 0;
const subaccount_1 = require("../subaccounts/subaccount");
const share_1 = require("./share");
const params_1 = require("./params");
const vault_1 = require("./vault");
const _m0 = __importStar(require("protobufjs/minimal"));
function createBaseMsgDepositToMegavault() {
    return {
        subaccountId: undefined,
        quoteQuantums: new Uint8Array()
    };
}
exports.MsgDepositToMegavault = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.subaccountId !== undefined) {
            subaccount_1.SubaccountId.encode(message.subaccountId, writer.uint32(10).fork()).ldelim();
        }
        if (message.quoteQuantums.length !== 0) {
            writer.uint32(18).bytes(message.quoteQuantums);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgDepositToMegavault();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.subaccountId = subaccount_1.SubaccountId.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.quoteQuantums = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseMsgDepositToMegavault();
        message.subaccountId = object.subaccountId !== undefined && object.subaccountId !== null ? subaccount_1.SubaccountId.fromPartial(object.subaccountId) : undefined;
        message.quoteQuantums = (_a = object.quoteQuantums) !== null && _a !== void 0 ? _a : new Uint8Array();
        return message;
    }
};
function createBaseMsgDepositToMegavaultResponse() {
    return {
        mintedShares: undefined
    };
}
exports.MsgDepositToMegavaultResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.mintedShares !== undefined) {
            share_1.NumShares.encode(message.mintedShares, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgDepositToMegavaultResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.mintedShares = share_1.NumShares.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseMsgDepositToMegavaultResponse();
        message.mintedShares = object.mintedShares !== undefined && object.mintedShares !== null ? share_1.NumShares.fromPartial(object.mintedShares) : undefined;
        return message;
    }
};
function createBaseMsgWithdrawFromMegavault() {
    return {
        subaccountId: undefined,
        shares: undefined,
        minQuoteQuantums: new Uint8Array()
    };
}
exports.MsgWithdrawFromMegavault = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.subaccountId !== undefined) {
            subaccount_1.SubaccountId.encode(message.subaccountId, writer.uint32(10).fork()).ldelim();
        }
        if (message.shares !== undefined) {
            share_1.NumShares.encode(message.shares, writer.uint32(18).fork()).ldelim();
        }
        if (message.minQuoteQuantums.length !== 0) {
            writer.uint32(26).bytes(message.minQuoteQuantums);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgWithdrawFromMegavault();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.subaccountId = subaccount_1.SubaccountId.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.shares = share_1.NumShares.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.minQuoteQuantums = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseMsgWithdrawFromMegavault();
        message.subaccountId = object.subaccountId !== undefined && object.subaccountId !== null ? subaccount_1.SubaccountId.fromPartial(object.subaccountId) : undefined;
        message.shares = object.shares !== undefined && object.shares !== null ? share_1.NumShares.fromPartial(object.shares) : undefined;
        message.minQuoteQuantums = (_a = object.minQuoteQuantums) !== null && _a !== void 0 ? _a : new Uint8Array();
        return message;
    }
};
function createBaseMsgWithdrawFromMegavaultResponse() {
    return {
        quoteQuantums: new Uint8Array()
    };
}
exports.MsgWithdrawFromMegavaultResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.quoteQuantums.length !== 0) {
            writer.uint32(10).bytes(message.quoteQuantums);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgWithdrawFromMegavaultResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.quoteQuantums = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseMsgWithdrawFromMegavaultResponse();
        message.quoteQuantums = (_a = object.quoteQuantums) !== null && _a !== void 0 ? _a : new Uint8Array();
        return message;
    }
};
function createBaseMsgUpdateDefaultQuotingParams() {
    return {
        authority: "",
        defaultQuotingParams: undefined
    };
}
exports.MsgUpdateDefaultQuotingParams = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.defaultQuotingParams !== undefined) {
            params_1.QuotingParams.encode(message.defaultQuotingParams, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateDefaultQuotingParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.defaultQuotingParams = params_1.QuotingParams.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseMsgUpdateDefaultQuotingParams();
        message.authority = (_a = object.authority) !== null && _a !== void 0 ? _a : "";
        message.defaultQuotingParams = object.defaultQuotingParams !== undefined && object.defaultQuotingParams !== null ? params_1.QuotingParams.fromPartial(object.defaultQuotingParams) : undefined;
        return message;
    }
};
function createBaseMsgUpdateDefaultQuotingParamsResponse() {
    return {};
}
exports.MsgUpdateDefaultQuotingParamsResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateDefaultQuotingParamsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgUpdateDefaultQuotingParamsResponse();
        return message;
    }
};
function createBaseMsgSetVaultParams() {
    return {
        authority: "",
        vaultId: undefined,
        vaultParams: undefined
    };
}
exports.MsgSetVaultParams = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.vaultId !== undefined) {
            vault_1.VaultId.encode(message.vaultId, writer.uint32(18).fork()).ldelim();
        }
        if (message.vaultParams !== undefined) {
            params_1.VaultParams.encode(message.vaultParams, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetVaultParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.vaultId = vault_1.VaultId.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.vaultParams = params_1.VaultParams.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseMsgSetVaultParams();
        message.authority = (_a = object.authority) !== null && _a !== void 0 ? _a : "";
        message.vaultId = object.vaultId !== undefined && object.vaultId !== null ? vault_1.VaultId.fromPartial(object.vaultId) : undefined;
        message.vaultParams = object.vaultParams !== undefined && object.vaultParams !== null ? params_1.VaultParams.fromPartial(object.vaultParams) : undefined;
        return message;
    }
};
function createBaseMsgSetVaultParamsResponse() {
    return {};
}
exports.MsgSetVaultParamsResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetVaultParamsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgSetVaultParamsResponse();
        return message;
    }
};
function createBaseMsgUnlockShares() {
    return {
        authority: "",
        ownerAddress: ""
    };
}
exports.MsgUnlockShares = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.ownerAddress !== "") {
            writer.uint32(18).string(message.ownerAddress);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUnlockShares();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.ownerAddress = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseMsgUnlockShares();
        message.authority = (_a = object.authority) !== null && _a !== void 0 ? _a : "";
        message.ownerAddress = (_b = object.ownerAddress) !== null && _b !== void 0 ? _b : "";
        return message;
    }
};
function createBaseMsgUnlockSharesResponse() {
    return {
        unlockedShares: undefined
    };
}
exports.MsgUnlockSharesResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.unlockedShares !== undefined) {
            share_1.NumShares.encode(message.unlockedShares, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUnlockSharesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.unlockedShares = share_1.NumShares.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseMsgUnlockSharesResponse();
        message.unlockedShares = object.unlockedShares !== undefined && object.unlockedShares !== null ? share_1.NumShares.fromPartial(object.unlockedShares) : undefined;
        return message;
    }
};
function createBaseMsgUpdateOperatorParams() {
    return {
        authority: "",
        params: undefined
    };
}
exports.MsgUpdateOperatorParams = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.params !== undefined) {
            params_1.OperatorParams.encode(message.params, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateOperatorParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.params = params_1.OperatorParams.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseMsgUpdateOperatorParams();
        message.authority = (_a = object.authority) !== null && _a !== void 0 ? _a : "";
        message.params = object.params !== undefined && object.params !== null ? params_1.OperatorParams.fromPartial(object.params) : undefined;
        return message;
    }
};
function createBaseMsgUpdateOperatorParamsResponse() {
    return {};
}
exports.MsgUpdateOperatorParamsResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateOperatorParamsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgUpdateOperatorParamsResponse();
        return message;
    }
};
function createBaseMsgAllocateToVault() {
    return {
        authority: "",
        vaultId: undefined,
        quoteQuantums: new Uint8Array()
    };
}
exports.MsgAllocateToVault = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.vaultId !== undefined) {
            vault_1.VaultId.encode(message.vaultId, writer.uint32(18).fork()).ldelim();
        }
        if (message.quoteQuantums.length !== 0) {
            writer.uint32(26).bytes(message.quoteQuantums);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgAllocateToVault();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.vaultId = vault_1.VaultId.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.quoteQuantums = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseMsgAllocateToVault();
        message.authority = (_a = object.authority) !== null && _a !== void 0 ? _a : "";
        message.vaultId = object.vaultId !== undefined && object.vaultId !== null ? vault_1.VaultId.fromPartial(object.vaultId) : undefined;
        message.quoteQuantums = (_b = object.quoteQuantums) !== null && _b !== void 0 ? _b : new Uint8Array();
        return message;
    }
};
function createBaseMsgAllocateToVaultResponse() {
    return {};
}
exports.MsgAllocateToVaultResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgAllocateToVaultResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgAllocateToVaultResponse();
        return message;
    }
};
function createBaseMsgRetrieveFromVault() {
    return {
        authority: "",
        vaultId: undefined,
        quoteQuantums: new Uint8Array()
    };
}
exports.MsgRetrieveFromVault = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.vaultId !== undefined) {
            vault_1.VaultId.encode(message.vaultId, writer.uint32(18).fork()).ldelim();
        }
        if (message.quoteQuantums.length !== 0) {
            writer.uint32(26).bytes(message.quoteQuantums);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgRetrieveFromVault();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.vaultId = vault_1.VaultId.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.quoteQuantums = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseMsgRetrieveFromVault();
        message.authority = (_a = object.authority) !== null && _a !== void 0 ? _a : "";
        message.vaultId = object.vaultId !== undefined && object.vaultId !== null ? vault_1.VaultId.fromPartial(object.vaultId) : undefined;
        message.quoteQuantums = (_b = object.quoteQuantums) !== null && _b !== void 0 ? _b : new Uint8Array();
        return message;
    }
};
function createBaseMsgRetrieveFromVaultResponse() {
    return {};
}
exports.MsgRetrieveFromVaultResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgRetrieveFromVaultResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgRetrieveFromVaultResponse();
        return message;
    }
};
function createBaseMsgUpdateParams() {
    return {
        authority: "",
        params: undefined
    };
}
exports.MsgUpdateParams = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.params !== undefined) {
            params_1.Params.encode(message.params, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.params = params_1.Params.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseMsgUpdateParams();
        message.authority = (_a = object.authority) !== null && _a !== void 0 ? _a : "";
        message.params = object.params !== undefined && object.params !== null ? params_1.Params.fromPartial(object.params) : undefined;
        return message;
    }
};
function createBaseMsgSetVaultQuotingParams() {
    return {
        authority: "",
        vaultId: undefined,
        quotingParams: undefined
    };
}
exports.MsgSetVaultQuotingParams = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.vaultId !== undefined) {
            vault_1.VaultId.encode(message.vaultId, writer.uint32(18).fork()).ldelim();
        }
        if (message.quotingParams !== undefined) {
            params_1.QuotingParams.encode(message.quotingParams, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetVaultQuotingParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.vaultId = vault_1.VaultId.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.quotingParams = params_1.QuotingParams.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseMsgSetVaultQuotingParams();
        message.authority = (_a = object.authority) !== null && _a !== void 0 ? _a : "";
        message.vaultId = object.vaultId !== undefined && object.vaultId !== null ? vault_1.VaultId.fromPartial(object.vaultId) : undefined;
        message.quotingParams = object.quotingParams !== undefined && object.quotingParams !== null ? params_1.QuotingParams.fromPartial(object.quotingParams) : undefined;
        return message;
    }
};
