// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/vault/tx.rpc.msg.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgClientImpl = void 0;
const _m0 = __importStar(require("protobufjs/minimal"));
const tx_1 = require("./tx");
class MsgClientImpl {
    constructor(rpc) {
        this.rpc = rpc;
        this.depositToMegavault = this.depositToMegavault.bind(this);
        this.withdrawFromMegavault = this.withdrawFromMegavault.bind(this);
        this.updateDefaultQuotingParams = this.updateDefaultQuotingParams.bind(this);
        this.updateOperatorParams = this.updateOperatorParams.bind(this);
        this.setVaultParams = this.setVaultParams.bind(this);
        this.unlockShares = this.unlockShares.bind(this);
        this.allocateToVault = this.allocateToVault.bind(this);
        this.retrieveFromVault = this.retrieveFromVault.bind(this);
    }
    depositToMegavault(request) {
        const data = tx_1.MsgDepositToMegavault.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.vault.Msg", "DepositToMegavault", data);
        return promise.then(data => tx_1.MsgDepositToMegavaultResponse.decode(new _m0.Reader(data)));
    }
    withdrawFromMegavault(request) {
        const data = tx_1.MsgWithdrawFromMegavault.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.vault.Msg", "WithdrawFromMegavault", data);
        return promise.then(data => tx_1.MsgWithdrawFromMegavaultResponse.decode(new _m0.Reader(data)));
    }
    updateDefaultQuotingParams(request) {
        const data = tx_1.MsgUpdateDefaultQuotingParams.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.vault.Msg", "UpdateDefaultQuotingParams", data);
        return promise.then(data => tx_1.MsgUpdateDefaultQuotingParamsResponse.decode(new _m0.Reader(data)));
    }
    updateOperatorParams(request) {
        const data = tx_1.MsgUpdateOperatorParams.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.vault.Msg", "UpdateOperatorParams", data);
        return promise.then(data => tx_1.MsgUpdateOperatorParamsResponse.decode(new _m0.Reader(data)));
    }
    setVaultParams(request) {
        const data = tx_1.MsgSetVaultParams.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.vault.Msg", "SetVaultParams", data);
        return promise.then(data => tx_1.MsgSetVaultParamsResponse.decode(new _m0.Reader(data)));
    }
    unlockShares(request) {
        const data = tx_1.MsgUnlockShares.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.vault.Msg", "UnlockShares", data);
        return promise.then(data => tx_1.MsgUnlockSharesResponse.decode(new _m0.Reader(data)));
    }
    allocateToVault(request) {
        const data = tx_1.MsgAllocateToVault.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.vault.Msg", "AllocateToVault", data);
        return promise.then(data => tx_1.MsgAllocateToVaultResponse.decode(new _m0.Reader(data)));
    }
    retrieveFromVault(request) {
        const data = tx_1.MsgRetrieveFromVault.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.vault.Msg", "RetrieveFromVault", data);
        return promise.then(data => tx_1.MsgRetrieveFromVaultResponse.decode(new _m0.Reader(data)));
    }
}
exports.MsgClientImpl = MsgClientImpl;
