// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/revshare/params.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MarketMapperRevenueShareParams = void 0;
const _m0 = __importStar(require("protobufjs/minimal"));
function createBaseMarketMapperRevenueShareParams() {
    return {
        address: "",
        revenueSharePpm: 0,
        validDays: 0
    };
}
exports.MarketMapperRevenueShareParams = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        if (message.revenueSharePpm !== 0) {
            writer.uint32(16).uint32(message.revenueSharePpm);
        }
        if (message.validDays !== 0) {
            writer.uint32(24).uint32(message.validDays);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMarketMapperRevenueShareParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.address = reader.string();
                    break;
                case 2:
                    message.revenueSharePpm = reader.uint32();
                    break;
                case 3:
                    message.validDays = reader.uint32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseMarketMapperRevenueShareParams();
        message.address = (_a = object.address) !== null && _a !== void 0 ? _a : "";
        message.revenueSharePpm = (_b = object.revenueSharePpm) !== null && _b !== void 0 ? _b : 0;
        message.validDays = (_c = object.validDays) !== null && _c !== void 0 ? _c : 0;
        return message;
    }
};
