// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/revshare/query.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryUnconditionalRevShareConfigResponse = exports.QueryUnconditionalRevShareConfig = exports.QueryMarketMapperRevShareDetailsResponse = exports.QueryMarketMapperRevShareDetails = exports.QueryMarketMapperRevenueShareParamsResponse = exports.QueryMarketMapperRevenueShareParams = void 0;
const params_1 = require("./params");
const revshare_1 = require("./revshare");
const _m0 = __importStar(require("protobufjs/minimal"));
function createBaseQueryMarketMapperRevenueShareParams() {
    return {};
}
exports.QueryMarketMapperRevenueShareParams = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryMarketMapperRevenueShareParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseQueryMarketMapperRevenueShareParams();
        return message;
    }
};
function createBaseQueryMarketMapperRevenueShareParamsResponse() {
    return {
        params: undefined
    };
}
exports.QueryMarketMapperRevenueShareParamsResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.params !== undefined) {
            params_1.MarketMapperRevenueShareParams.encode(message.params, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryMarketMapperRevenueShareParamsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.params = params_1.MarketMapperRevenueShareParams.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseQueryMarketMapperRevenueShareParamsResponse();
        message.params = object.params !== undefined && object.params !== null ? params_1.MarketMapperRevenueShareParams.fromPartial(object.params) : undefined;
        return message;
    }
};
function createBaseQueryMarketMapperRevShareDetails() {
    return {
        marketId: 0
    };
}
exports.QueryMarketMapperRevShareDetails = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.marketId !== 0) {
            writer.uint32(8).uint32(message.marketId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryMarketMapperRevShareDetails();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.marketId = reader.uint32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseQueryMarketMapperRevShareDetails();
        message.marketId = (_a = object.marketId) !== null && _a !== void 0 ? _a : 0;
        return message;
    }
};
function createBaseQueryMarketMapperRevShareDetailsResponse() {
    return {
        details: undefined
    };
}
exports.QueryMarketMapperRevShareDetailsResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.details !== undefined) {
            revshare_1.MarketMapperRevShareDetails.encode(message.details, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryMarketMapperRevShareDetailsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.details = revshare_1.MarketMapperRevShareDetails.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseQueryMarketMapperRevShareDetailsResponse();
        message.details = object.details !== undefined && object.details !== null ? revshare_1.MarketMapperRevShareDetails.fromPartial(object.details) : undefined;
        return message;
    }
};
function createBaseQueryUnconditionalRevShareConfig() {
    return {};
}
exports.QueryUnconditionalRevShareConfig = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryUnconditionalRevShareConfig();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseQueryUnconditionalRevShareConfig();
        return message;
    }
};
function createBaseQueryUnconditionalRevShareConfigResponse() {
    return {
        config: undefined
    };
}
exports.QueryUnconditionalRevShareConfigResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.config !== undefined) {
            revshare_1.UnconditionalRevShareConfig.encode(message.config, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryUnconditionalRevShareConfigResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.config = revshare_1.UnconditionalRevShareConfig.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseQueryUnconditionalRevShareConfigResponse();
        message.config = object.config !== undefined && object.config !== null ? revshare_1.UnconditionalRevShareConfig.fromPartial(object.config) : undefined;
        return message;
    }
};
