// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/revshare/revshare.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnconditionalRevShareConfig_RecipientConfig = exports.UnconditionalRevShareConfig = exports.MarketMapperRevShareDetails = void 0;
const helpers_1 = require("../../helpers");
const _m0 = __importStar(require("protobufjs/minimal"));
function createBaseMarketMapperRevShareDetails() {
    return {
        expirationTs: helpers_1.Long.UZERO
    };
}
exports.MarketMapperRevShareDetails = {
    encode(message, writer = _m0.Writer.create()) {
        if (!message.expirationTs.isZero()) {
            writer.uint32(8).uint64(message.expirationTs);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMarketMapperRevShareDetails();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.expirationTs = reader.uint64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseMarketMapperRevShareDetails();
        message.expirationTs = object.expirationTs !== undefined && object.expirationTs !== null ? helpers_1.Long.fromValue(object.expirationTs) : helpers_1.Long.UZERO;
        return message;
    }
};
function createBaseUnconditionalRevShareConfig() {
    return {
        configs: []
    };
}
exports.UnconditionalRevShareConfig = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.configs) {
            exports.UnconditionalRevShareConfig_RecipientConfig.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUnconditionalRevShareConfig();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.configs.push(exports.UnconditionalRevShareConfig_RecipientConfig.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseUnconditionalRevShareConfig();
        message.configs = ((_a = object.configs) === null || _a === void 0 ? void 0 : _a.map(e => exports.UnconditionalRevShareConfig_RecipientConfig.fromPartial(e))) || [];
        return message;
    }
};
function createBaseUnconditionalRevShareConfig_RecipientConfig() {
    return {
        address: "",
        sharePpm: 0
    };
}
exports.UnconditionalRevShareConfig_RecipientConfig = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        if (message.sharePpm !== 0) {
            writer.uint32(16).uint32(message.sharePpm);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUnconditionalRevShareConfig_RecipientConfig();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.address = reader.string();
                    break;
                case 2:
                    message.sharePpm = reader.uint32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseUnconditionalRevShareConfig_RecipientConfig();
        message.address = (_a = object.address) !== null && _a !== void 0 ? _a : "";
        message.sharePpm = (_b = object.sharePpm) !== null && _b !== void 0 ? _b : 0;
        return message;
    }
};
