// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/revshare/tx.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgUpdateUnconditionalRevShareConfigResponse = exports.MsgUpdateUnconditionalRevShareConfig = exports.MsgSetMarketMapperRevShareDetailsForMarketResponse = exports.MsgSetMarketMapperRevShareDetailsForMarket = exports.MsgSetMarketMapperRevenueShareResponse = exports.MsgSetMarketMapperRevenueShare = void 0;
const params_1 = require("./params");
const revshare_1 = require("./revshare");
const _m0 = __importStar(require("protobufjs/minimal"));
function createBaseMsgSetMarketMapperRevenueShare() {
    return {
        authority: "",
        params: undefined
    };
}
exports.MsgSetMarketMapperRevenueShare = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.params !== undefined) {
            params_1.MarketMapperRevenueShareParams.encode(message.params, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetMarketMapperRevenueShare();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.params = params_1.MarketMapperRevenueShareParams.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseMsgSetMarketMapperRevenueShare();
        message.authority = (_a = object.authority) !== null && _a !== void 0 ? _a : "";
        message.params = object.params !== undefined && object.params !== null ? params_1.MarketMapperRevenueShareParams.fromPartial(object.params) : undefined;
        return message;
    }
};
function createBaseMsgSetMarketMapperRevenueShareResponse() {
    return {};
}
exports.MsgSetMarketMapperRevenueShareResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetMarketMapperRevenueShareResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgSetMarketMapperRevenueShareResponse();
        return message;
    }
};
function createBaseMsgSetMarketMapperRevShareDetailsForMarket() {
    return {
        authority: "",
        marketId: 0,
        params: undefined
    };
}
exports.MsgSetMarketMapperRevShareDetailsForMarket = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.marketId !== 0) {
            writer.uint32(16).uint32(message.marketId);
        }
        if (message.params !== undefined) {
            revshare_1.MarketMapperRevShareDetails.encode(message.params, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetMarketMapperRevShareDetailsForMarket();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.marketId = reader.uint32();
                    break;
                case 3:
                    message.params = revshare_1.MarketMapperRevShareDetails.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseMsgSetMarketMapperRevShareDetailsForMarket();
        message.authority = (_a = object.authority) !== null && _a !== void 0 ? _a : "";
        message.marketId = (_b = object.marketId) !== null && _b !== void 0 ? _b : 0;
        message.params = object.params !== undefined && object.params !== null ? revshare_1.MarketMapperRevShareDetails.fromPartial(object.params) : undefined;
        return message;
    }
};
function createBaseMsgSetMarketMapperRevShareDetailsForMarketResponse() {
    return {};
}
exports.MsgSetMarketMapperRevShareDetailsForMarketResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetMarketMapperRevShareDetailsForMarketResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgSetMarketMapperRevShareDetailsForMarketResponse();
        return message;
    }
};
function createBaseMsgUpdateUnconditionalRevShareConfig() {
    return {
        authority: "",
        config: undefined
    };
}
exports.MsgUpdateUnconditionalRevShareConfig = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.config !== undefined) {
            revshare_1.UnconditionalRevShareConfig.encode(message.config, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateUnconditionalRevShareConfig();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.config = revshare_1.UnconditionalRevShareConfig.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseMsgUpdateUnconditionalRevShareConfig();
        message.authority = (_a = object.authority) !== null && _a !== void 0 ? _a : "";
        message.config = object.config !== undefined && object.config !== null ? revshare_1.UnconditionalRevShareConfig.fromPartial(object.config) : undefined;
        return message;
    }
};
function createBaseMsgUpdateUnconditionalRevShareConfigResponse() {
    return {};
}
exports.MsgUpdateUnconditionalRevShareConfigResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpdateUnconditionalRevShareConfigResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgUpdateUnconditionalRevShareConfigResponse();
        return message;
    }
};
