// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/ratelimit/query.rpc.Query.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createRpcQueryExtension = exports.QueryClientImpl = void 0;
const _m0 = __importStar(require("protobufjs/minimal"));
const stargate_1 = require("@cosmjs/stargate");
const query_1 = require("./query");
class QueryClientImpl {
    constructor(rpc) {
        this.rpc = rpc;
        this.listLimitParams = this.listLimitParams.bind(this);
        this.capacityByDenom = this.capacityByDenom.bind(this);
        this.allPendingSendPackets = this.allPendingSendPackets.bind(this);
    }
    listLimitParams(request = {}) {
        const data = query_1.ListLimitParamsRequest.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.ratelimit.Query", "ListLimitParams", data);
        return promise.then(data => query_1.ListLimitParamsResponse.decode(new _m0.Reader(data)));
    }
    capacityByDenom(request) {
        const data = query_1.QueryCapacityByDenomRequest.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.ratelimit.Query", "CapacityByDenom", data);
        return promise.then(data => query_1.QueryCapacityByDenomResponse.decode(new _m0.Reader(data)));
    }
    allPendingSendPackets(request = {}) {
        const data = query_1.QueryAllPendingSendPacketsRequest.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.ratelimit.Query", "AllPendingSendPackets", data);
        return promise.then(data => query_1.QueryAllPendingSendPacketsResponse.decode(new _m0.Reader(data)));
    }
}
exports.QueryClientImpl = QueryClientImpl;
const createRpcQueryExtension = (base) => {
    const rpc = (0, stargate_1.createProtobufRpcClient)(base);
    const queryService = new QueryClientImpl(rpc);
    return {
        listLimitParams(request) {
            return queryService.listLimitParams(request);
        },
        capacityByDenom(request) {
            return queryService.capacityByDenom(request);
        },
        allPendingSendPackets(request) {
            return queryService.allPendingSendPackets(request);
        }
    };
};
exports.createRpcQueryExtension = createRpcQueryExtension;
