// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/accountplus/accountplus.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimestampNonceDetails = exports.AccountState = void 0;
const _m0 = __importStar(require("protobufjs/minimal"));
const helpers_1 = require("../../helpers");
function createBaseAccountState() {
    return {
        address: "",
        timestampNonceDetails: undefined
    };
}
exports.AccountState = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        if (message.timestampNonceDetails !== undefined) {
            exports.TimestampNonceDetails.encode(message.timestampNonceDetails, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAccountState();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.address = reader.string();
                    break;
                case 2:
                    message.timestampNonceDetails = exports.TimestampNonceDetails.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseAccountState();
        message.address = (_a = object.address) !== null && _a !== void 0 ? _a : "";
        message.timestampNonceDetails = object.timestampNonceDetails !== undefined && object.timestampNonceDetails !== null ? exports.TimestampNonceDetails.fromPartial(object.timestampNonceDetails) : undefined;
        return message;
    }
};
function createBaseTimestampNonceDetails() {
    return {
        timestampNonces: [],
        maxEjectedNonce: helpers_1.Long.UZERO
    };
}
exports.TimestampNonceDetails = {
    encode(message, writer = _m0.Writer.create()) {
        writer.uint32(10).fork();
        for (const v of message.timestampNonces) {
            writer.uint64(v);
        }
        writer.ldelim();
        if (!message.maxEjectedNonce.isZero()) {
            writer.uint32(16).uint64(message.maxEjectedNonce);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseTimestampNonceDetails();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if ((tag & 7) === 2) {
                        const end2 = reader.uint32() + reader.pos;
                        while (reader.pos < end2) {
                            message.timestampNonces.push(reader.uint64());
                        }
                    }
                    else {
                        message.timestampNonces.push(reader.uint64());
                    }
                    break;
                case 2:
                    message.maxEjectedNonce = reader.uint64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseTimestampNonceDetails();
        message.timestampNonces = ((_a = object.timestampNonces) === null || _a === void 0 ? void 0 : _a.map(e => helpers_1.Long.fromValue(e))) || [];
        message.maxEjectedNonce = object.maxEjectedNonce !== undefined && object.maxEjectedNonce !== null ? helpers_1.Long.fromValue(object.maxEjectedNonce) : helpers_1.Long.UZERO;
        return message;
    }
};
