// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/accountplus/query.rpc.Query.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createRpcQueryExtension = exports.QueryClientImpl = void 0;
const _m0 = __importStar(require("protobufjs/minimal"));
const stargate_1 = require("@cosmjs/stargate");
const query_1 = require("./query");
class QueryClientImpl {
    constructor(rpc) {
        this.rpc = rpc;
        this.params = this.params.bind(this);
        this.getAuthenticator = this.getAuthenticator.bind(this);
        this.getAuthenticators = this.getAuthenticators.bind(this);
    }
    params(request = {}) {
        const data = query_1.QueryParamsRequest.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.accountplus.Query", "Params", data);
        return promise.then(data => query_1.QueryParamsResponse.decode(new _m0.Reader(data)));
    }
    getAuthenticator(request) {
        const data = query_1.GetAuthenticatorRequest.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.accountplus.Query", "GetAuthenticator", data);
        return promise.then(data => query_1.GetAuthenticatorResponse.decode(new _m0.Reader(data)));
    }
    getAuthenticators(request) {
        const data = query_1.GetAuthenticatorsRequest.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.accountplus.Query", "GetAuthenticators", data);
        return promise.then(data => query_1.GetAuthenticatorsResponse.decode(new _m0.Reader(data)));
    }
}
exports.QueryClientImpl = QueryClientImpl;
const createRpcQueryExtension = (base) => {
    const rpc = (0, stargate_1.createProtobufRpcClient)(base);
    const queryService = new QueryClientImpl(rpc);
    return {
        params(request) {
            return queryService.params(request);
        },
        getAuthenticator(request) {
            return queryService.getAuthenticator(request);
        },
        getAuthenticators(request) {
            return queryService.getAuthenticators(request);
        }
    };
};
exports.createRpcQueryExtension = createRpcQueryExtension;
