// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/subaccounts/query.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryCollateralPoolAddressResponse = exports.QueryCollateralPoolAddressRequest = exports.QueryGetWithdrawalAndTransfersBlockedInfoResponse = exports.QueryGetWithdrawalAndTransfersBlockedInfoRequest = exports.QuerySubaccountAllResponse = exports.QueryAllSubaccountRequest = exports.QuerySubaccountResponse = exports.QueryGetSubaccountRequest = void 0;
const pagination_1 = require("../../cosmos/base/query/v1beta1/pagination");
const subaccount_1 = require("./subaccount");
const _m0 = __importStar(require("protobufjs/minimal"));
function createBaseQueryGetSubaccountRequest() {
    return {
        owner: "",
        number: 0
    };
}
exports.QueryGetSubaccountRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.owner !== "") {
            writer.uint32(10).string(message.owner);
        }
        if (message.number !== 0) {
            writer.uint32(16).uint32(message.number);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryGetSubaccountRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.owner = reader.string();
                    break;
                case 2:
                    message.number = reader.uint32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseQueryGetSubaccountRequest();
        message.owner = (_a = object.owner) !== null && _a !== void 0 ? _a : "";
        message.number = (_b = object.number) !== null && _b !== void 0 ? _b : 0;
        return message;
    }
};
function createBaseQuerySubaccountResponse() {
    return {
        subaccount: undefined
    };
}
exports.QuerySubaccountResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.subaccount !== undefined) {
            subaccount_1.Subaccount.encode(message.subaccount, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQuerySubaccountResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.subaccount = subaccount_1.Subaccount.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseQuerySubaccountResponse();
        message.subaccount = object.subaccount !== undefined && object.subaccount !== null ? subaccount_1.Subaccount.fromPartial(object.subaccount) : undefined;
        return message;
    }
};
function createBaseQueryAllSubaccountRequest() {
    return {
        pagination: undefined
    };
}
exports.QueryAllSubaccountRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.pagination !== undefined) {
            pagination_1.PageRequest.encode(message.pagination, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryAllSubaccountRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.pagination = pagination_1.PageRequest.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseQueryAllSubaccountRequest();
        message.pagination = object.pagination !== undefined && object.pagination !== null ? pagination_1.PageRequest.fromPartial(object.pagination) : undefined;
        return message;
    }
};
function createBaseQuerySubaccountAllResponse() {
    return {
        subaccount: [],
        pagination: undefined
    };
}
exports.QuerySubaccountAllResponse = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.subaccount) {
            subaccount_1.Subaccount.encode(v, writer.uint32(10).fork()).ldelim();
        }
        if (message.pagination !== undefined) {
            pagination_1.PageResponse.encode(message.pagination, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQuerySubaccountAllResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.subaccount.push(subaccount_1.Subaccount.decode(reader, reader.uint32()));
                    break;
                case 2:
                    message.pagination = pagination_1.PageResponse.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseQuerySubaccountAllResponse();
        message.subaccount = ((_a = object.subaccount) === null || _a === void 0 ? void 0 : _a.map(e => subaccount_1.Subaccount.fromPartial(e))) || [];
        message.pagination = object.pagination !== undefined && object.pagination !== null ? pagination_1.PageResponse.fromPartial(object.pagination) : undefined;
        return message;
    }
};
function createBaseQueryGetWithdrawalAndTransfersBlockedInfoRequest() {
    return {
        perpetualId: 0
    };
}
exports.QueryGetWithdrawalAndTransfersBlockedInfoRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.perpetualId !== 0) {
            writer.uint32(8).uint32(message.perpetualId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryGetWithdrawalAndTransfersBlockedInfoRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.perpetualId = reader.uint32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseQueryGetWithdrawalAndTransfersBlockedInfoRequest();
        message.perpetualId = (_a = object.perpetualId) !== null && _a !== void 0 ? _a : 0;
        return message;
    }
};
function createBaseQueryGetWithdrawalAndTransfersBlockedInfoResponse() {
    return {
        negativeTncSubaccountSeenAtBlock: 0,
        chainOutageSeenAtBlock: 0,
        withdrawalsAndTransfersUnblockedAtBlock: 0
    };
}
exports.QueryGetWithdrawalAndTransfersBlockedInfoResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.negativeTncSubaccountSeenAtBlock !== 0) {
            writer.uint32(8).uint32(message.negativeTncSubaccountSeenAtBlock);
        }
        if (message.chainOutageSeenAtBlock !== 0) {
            writer.uint32(16).uint32(message.chainOutageSeenAtBlock);
        }
        if (message.withdrawalsAndTransfersUnblockedAtBlock !== 0) {
            writer.uint32(24).uint32(message.withdrawalsAndTransfersUnblockedAtBlock);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryGetWithdrawalAndTransfersBlockedInfoResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.negativeTncSubaccountSeenAtBlock = reader.uint32();
                    break;
                case 2:
                    message.chainOutageSeenAtBlock = reader.uint32();
                    break;
                case 3:
                    message.withdrawalsAndTransfersUnblockedAtBlock = reader.uint32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseQueryGetWithdrawalAndTransfersBlockedInfoResponse();
        message.negativeTncSubaccountSeenAtBlock = (_a = object.negativeTncSubaccountSeenAtBlock) !== null && _a !== void 0 ? _a : 0;
        message.chainOutageSeenAtBlock = (_b = object.chainOutageSeenAtBlock) !== null && _b !== void 0 ? _b : 0;
        message.withdrawalsAndTransfersUnblockedAtBlock = (_c = object.withdrawalsAndTransfersUnblockedAtBlock) !== null && _c !== void 0 ? _c : 0;
        return message;
    }
};
function createBaseQueryCollateralPoolAddressRequest() {
    return {
        perpetualId: 0
    };
}
exports.QueryCollateralPoolAddressRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.perpetualId !== 0) {
            writer.uint32(8).uint32(message.perpetualId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryCollateralPoolAddressRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.perpetualId = reader.uint32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseQueryCollateralPoolAddressRequest();
        message.perpetualId = (_a = object.perpetualId) !== null && _a !== void 0 ? _a : 0;
        return message;
    }
};
function createBaseQueryCollateralPoolAddressResponse() {
    return {
        collateralPoolAddress: ""
    };
}
exports.QueryCollateralPoolAddressResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.collateralPoolAddress !== "") {
            writer.uint32(10).string(message.collateralPoolAddress);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryCollateralPoolAddressResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.collateralPoolAddress = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseQueryCollateralPoolAddressResponse();
        message.collateralPoolAddress = (_a = object.collateralPoolAddress) !== null && _a !== void 0 ? _a : "";
        return message;
    }
};
