// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/subaccounts/streaming.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubaccountAssetPosition = exports.SubaccountPerpetualPosition = exports.StreamSubaccountUpdate = void 0;
const subaccount_1 = require("./subaccount");
const _m0 = __importStar(require("protobufjs/minimal"));
const helpers_1 = require("../../helpers");
function createBaseStreamSubaccountUpdate() {
    return {
        subaccountId: undefined,
        updatedPerpetualPositions: [],
        updatedAssetPositions: [],
        snapshot: false
    };
}
exports.StreamSubaccountUpdate = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.subaccountId !== undefined) {
            subaccount_1.SubaccountId.encode(message.subaccountId, writer.uint32(10).fork()).ldelim();
        }
        for (const v of message.updatedPerpetualPositions) {
            exports.SubaccountPerpetualPosition.encode(v, writer.uint32(18).fork()).ldelim();
        }
        for (const v of message.updatedAssetPositions) {
            exports.SubaccountAssetPosition.encode(v, writer.uint32(26).fork()).ldelim();
        }
        if (message.snapshot === true) {
            writer.uint32(32).bool(message.snapshot);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseStreamSubaccountUpdate();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.subaccountId = subaccount_1.SubaccountId.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.updatedPerpetualPositions.push(exports.SubaccountPerpetualPosition.decode(reader, reader.uint32()));
                    break;
                case 3:
                    message.updatedAssetPositions.push(exports.SubaccountAssetPosition.decode(reader, reader.uint32()));
                    break;
                case 4:
                    message.snapshot = reader.bool();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseStreamSubaccountUpdate();
        message.subaccountId = object.subaccountId !== undefined && object.subaccountId !== null ? subaccount_1.SubaccountId.fromPartial(object.subaccountId) : undefined;
        message.updatedPerpetualPositions = ((_a = object.updatedPerpetualPositions) === null || _a === void 0 ? void 0 : _a.map(e => exports.SubaccountPerpetualPosition.fromPartial(e))) || [];
        message.updatedAssetPositions = ((_b = object.updatedAssetPositions) === null || _b === void 0 ? void 0 : _b.map(e => exports.SubaccountAssetPosition.fromPartial(e))) || [];
        message.snapshot = (_c = object.snapshot) !== null && _c !== void 0 ? _c : false;
        return message;
    }
};
function createBaseSubaccountPerpetualPosition() {
    return {
        perpetualId: 0,
        quantums: helpers_1.Long.ZERO
    };
}
exports.SubaccountPerpetualPosition = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.perpetualId !== 0) {
            writer.uint32(8).uint32(message.perpetualId);
        }
        if (!message.quantums.isZero()) {
            writer.uint32(16).int64(message.quantums);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSubaccountPerpetualPosition();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.perpetualId = reader.uint32();
                    break;
                case 2:
                    message.quantums = reader.int64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseSubaccountPerpetualPosition();
        message.perpetualId = (_a = object.perpetualId) !== null && _a !== void 0 ? _a : 0;
        message.quantums = object.quantums !== undefined && object.quantums !== null ? helpers_1.Long.fromValue(object.quantums) : helpers_1.Long.ZERO;
        return message;
    }
};
function createBaseSubaccountAssetPosition() {
    return {
        assetId: 0,
        quantums: helpers_1.Long.UZERO
    };
}
exports.SubaccountAssetPosition = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.assetId !== 0) {
            writer.uint32(8).uint32(message.assetId);
        }
        if (!message.quantums.isZero()) {
            writer.uint32(16).uint64(message.quantums);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSubaccountAssetPosition();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.assetId = reader.uint32();
                    break;
                case 2:
                    message.quantums = reader.uint64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseSubaccountAssetPosition();
        message.assetId = (_a = object.assetId) !== null && _a !== void 0 ? _a : 0;
        message.quantums = object.quantums !== undefined && object.quantums !== null ? helpers_1.Long.fromValue(object.quantums) : helpers_1.Long.UZERO;
        return message;
    }
};
