// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/clob/tx.rpc.msg.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgClientImpl = void 0;
const _m0 = __importStar(require("protobufjs/minimal"));
const tx_1 = require("./tx");
class MsgClientImpl {
    constructor(rpc) {
        this.rpc = rpc;
        this.proposedOperations = this.proposedOperations.bind(this);
        this.placeOrder = this.placeOrder.bind(this);
        this.cancelOrder = this.cancelOrder.bind(this);
        this.batchCancel = this.batchCancel.bind(this);
        this.createClobPair = this.createClobPair.bind(this);
        this.updateClobPair = this.updateClobPair.bind(this);
        this.updateEquityTierLimitConfiguration = this.updateEquityTierLimitConfiguration.bind(this);
        this.updateBlockRateLimitConfiguration = this.updateBlockRateLimitConfiguration.bind(this);
        this.updateLiquidationsConfig = this.updateLiquidationsConfig.bind(this);
    }
    proposedOperations(request) {
        const data = tx_1.MsgProposedOperations.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.clob.Msg", "ProposedOperations", data);
        return promise.then(data => tx_1.MsgProposedOperationsResponse.decode(new _m0.Reader(data)));
    }
    placeOrder(request) {
        const data = tx_1.MsgPlaceOrder.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.clob.Msg", "PlaceOrder", data);
        return promise.then(data => tx_1.MsgPlaceOrderResponse.decode(new _m0.Reader(data)));
    }
    cancelOrder(request) {
        const data = tx_1.MsgCancelOrder.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.clob.Msg", "CancelOrder", data);
        return promise.then(data => tx_1.MsgCancelOrderResponse.decode(new _m0.Reader(data)));
    }
    batchCancel(request) {
        const data = tx_1.MsgBatchCancel.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.clob.Msg", "BatchCancel", data);
        return promise.then(data => tx_1.MsgBatchCancelResponse.decode(new _m0.Reader(data)));
    }
    createClobPair(request) {
        const data = tx_1.MsgCreateClobPair.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.clob.Msg", "CreateClobPair", data);
        return promise.then(data => tx_1.MsgCreateClobPairResponse.decode(new _m0.Reader(data)));
    }
    updateClobPair(request) {
        const data = tx_1.MsgUpdateClobPair.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.clob.Msg", "UpdateClobPair", data);
        return promise.then(data => tx_1.MsgUpdateClobPairResponse.decode(new _m0.Reader(data)));
    }
    updateEquityTierLimitConfiguration(request) {
        const data = tx_1.MsgUpdateEquityTierLimitConfiguration.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.clob.Msg", "UpdateEquityTierLimitConfiguration", data);
        return promise.then(data => tx_1.MsgUpdateEquityTierLimitConfigurationResponse.decode(new _m0.Reader(data)));
    }
    updateBlockRateLimitConfiguration(request) {
        const data = tx_1.MsgUpdateBlockRateLimitConfiguration.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.clob.Msg", "UpdateBlockRateLimitConfiguration", data);
        return promise.then(data => tx_1.MsgUpdateBlockRateLimitConfigurationResponse.decode(new _m0.Reader(data)));
    }
    updateLiquidationsConfig(request) {
        const data = tx_1.MsgUpdateLiquidationsConfig.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.clob.Msg", "UpdateLiquidationsConfig", data);
        return promise.then(data => tx_1.MsgUpdateLiquidationsConfigResponse.decode(new _m0.Reader(data)));
    }
}
exports.MsgClientImpl = MsgClientImpl;
