// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/indexer/events/events.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpsertVaultEventV1 = exports.RegisterAffiliateEventV1 = exports.LiquidityTierUpsertEventV2 = exports.OpenInterestUpdate = exports.OpenInterestUpdateEventV1 = exports.AddressTradingReward = exports.TradingRewardsEventV1 = exports.UpdatePerpetualEventV2 = exports.UpdatePerpetualEventV1 = exports.UpdateClobPairEventV1 = exports.LiquidityTierUpsertEventV1 = exports.PerpetualMarketCreateEventV2 = exports.PerpetualMarketCreateEventV1 = exports.AssetCreateEventV1 = exports.StatefulOrderEventV1_LongTermOrderReplacementV1 = exports.StatefulOrderEventV1_LongTermOrderPlacementV1 = exports.StatefulOrderEventV1_ConditionalOrderTriggeredV1 = exports.StatefulOrderEventV1_ConditionalOrderPlacementV1 = exports.StatefulOrderEventV1_StatefulOrderRemovalV1 = exports.StatefulOrderEventV1_StatefulOrderPlacementV1 = exports.StatefulOrderEventV1 = exports.SubaccountUpdateEventV1 = exports.LiquidationOrderV1 = exports.DeleveragingEventV1 = exports.OrderFillEventV1 = exports.TransferEventV1 = exports.SourceOfFunds = exports.MarketModifyEventV1 = exports.MarketCreateEventV1 = exports.MarketBaseEventV1 = exports.MarketPriceUpdateEventV1 = exports.MarketEventV1 = exports.FundingEventV1 = exports.FundingUpdateV1 = exports.fundingEventV1_TypeToJSON = exports.fundingEventV1_TypeFromJSON = exports.FundingEventV1_TypeSDKType = exports.FundingEventV1_Type = void 0;
const subaccount_1 = require("../protocol/v1/subaccount");
const clob_1 = require("../protocol/v1/clob");
const _m0 = __importStar(require("protobufjs/minimal"));
const helpers_1 = require("../../../helpers");
/** Type is the type for funding values. */
var FundingEventV1_Type;
(function (FundingEventV1_Type) {
    /** TYPE_UNSPECIFIED - Unspecified type. */
    FundingEventV1_Type[FundingEventV1_Type["TYPE_UNSPECIFIED"] = 0] = "TYPE_UNSPECIFIED";
    /**
     * TYPE_PREMIUM_SAMPLE - Premium sample is the combined value from all premium votes during a
     * `funding-sample` epoch.
     */
    FundingEventV1_Type[FundingEventV1_Type["TYPE_PREMIUM_SAMPLE"] = 1] = "TYPE_PREMIUM_SAMPLE";
    /**
     * TYPE_FUNDING_RATE_AND_INDEX - Funding rate is the final funding rate combining all premium samples
     * during a `funding-tick` epoch.
     */
    FundingEventV1_Type[FundingEventV1_Type["TYPE_FUNDING_RATE_AND_INDEX"] = 2] = "TYPE_FUNDING_RATE_AND_INDEX";
    /**
     * TYPE_PREMIUM_VOTE - TODO(DEC-1513): Investigate whether premium vote values need to be
     * sent to indexer.
     */
    FundingEventV1_Type[FundingEventV1_Type["TYPE_PREMIUM_VOTE"] = 3] = "TYPE_PREMIUM_VOTE";
    FundingEventV1_Type[FundingEventV1_Type["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(FundingEventV1_Type = exports.FundingEventV1_Type || (exports.FundingEventV1_Type = {}));
exports.FundingEventV1_TypeSDKType = FundingEventV1_Type;
function fundingEventV1_TypeFromJSON(object) {
    switch (object) {
        case 0:
        case "TYPE_UNSPECIFIED":
            return FundingEventV1_Type.TYPE_UNSPECIFIED;
        case 1:
        case "TYPE_PREMIUM_SAMPLE":
            return FundingEventV1_Type.TYPE_PREMIUM_SAMPLE;
        case 2:
        case "TYPE_FUNDING_RATE_AND_INDEX":
            return FundingEventV1_Type.TYPE_FUNDING_RATE_AND_INDEX;
        case 3:
        case "TYPE_PREMIUM_VOTE":
            return FundingEventV1_Type.TYPE_PREMIUM_VOTE;
        case -1:
        case "UNRECOGNIZED":
        default:
            return FundingEventV1_Type.UNRECOGNIZED;
    }
}
exports.fundingEventV1_TypeFromJSON = fundingEventV1_TypeFromJSON;
function fundingEventV1_TypeToJSON(object) {
    switch (object) {
        case FundingEventV1_Type.TYPE_UNSPECIFIED:
            return "TYPE_UNSPECIFIED";
        case FundingEventV1_Type.TYPE_PREMIUM_SAMPLE:
            return "TYPE_PREMIUM_SAMPLE";
        case FundingEventV1_Type.TYPE_FUNDING_RATE_AND_INDEX:
            return "TYPE_FUNDING_RATE_AND_INDEX";
        case FundingEventV1_Type.TYPE_PREMIUM_VOTE:
            return "TYPE_PREMIUM_VOTE";
        case FundingEventV1_Type.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
exports.fundingEventV1_TypeToJSON = fundingEventV1_TypeToJSON;
function createBaseFundingUpdateV1() {
    return {
        perpetualId: 0,
        fundingValuePpm: 0,
        fundingIndex: new Uint8Array()
    };
}
exports.FundingUpdateV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.perpetualId !== 0) {
            writer.uint32(8).uint32(message.perpetualId);
        }
        if (message.fundingValuePpm !== 0) {
            writer.uint32(16).int32(message.fundingValuePpm);
        }
        if (message.fundingIndex.length !== 0) {
            writer.uint32(26).bytes(message.fundingIndex);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseFundingUpdateV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.perpetualId = reader.uint32();
                    break;
                case 2:
                    message.fundingValuePpm = reader.int32();
                    break;
                case 3:
                    message.fundingIndex = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseFundingUpdateV1();
        message.perpetualId = (_a = object.perpetualId) !== null && _a !== void 0 ? _a : 0;
        message.fundingValuePpm = (_b = object.fundingValuePpm) !== null && _b !== void 0 ? _b : 0;
        message.fundingIndex = (_c = object.fundingIndex) !== null && _c !== void 0 ? _c : new Uint8Array();
        return message;
    }
};
function createBaseFundingEventV1() {
    return {
        updates: [],
        type: 0
    };
}
exports.FundingEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.updates) {
            exports.FundingUpdateV1.encode(v, writer.uint32(10).fork()).ldelim();
        }
        if (message.type !== 0) {
            writer.uint32(16).int32(message.type);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseFundingEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.updates.push(exports.FundingUpdateV1.decode(reader, reader.uint32()));
                    break;
                case 2:
                    message.type = reader.int32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseFundingEventV1();
        message.updates = ((_a = object.updates) === null || _a === void 0 ? void 0 : _a.map(e => exports.FundingUpdateV1.fromPartial(e))) || [];
        message.type = (_b = object.type) !== null && _b !== void 0 ? _b : 0;
        return message;
    }
};
function createBaseMarketEventV1() {
    return {
        marketId: 0,
        priceUpdate: undefined,
        marketCreate: undefined,
        marketModify: undefined
    };
}
exports.MarketEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.marketId !== 0) {
            writer.uint32(8).uint32(message.marketId);
        }
        if (message.priceUpdate !== undefined) {
            exports.MarketPriceUpdateEventV1.encode(message.priceUpdate, writer.uint32(18).fork()).ldelim();
        }
        if (message.marketCreate !== undefined) {
            exports.MarketCreateEventV1.encode(message.marketCreate, writer.uint32(26).fork()).ldelim();
        }
        if (message.marketModify !== undefined) {
            exports.MarketModifyEventV1.encode(message.marketModify, writer.uint32(34).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMarketEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.marketId = reader.uint32();
                    break;
                case 2:
                    message.priceUpdate = exports.MarketPriceUpdateEventV1.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.marketCreate = exports.MarketCreateEventV1.decode(reader, reader.uint32());
                    break;
                case 4:
                    message.marketModify = exports.MarketModifyEventV1.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseMarketEventV1();
        message.marketId = (_a = object.marketId) !== null && _a !== void 0 ? _a : 0;
        message.priceUpdate = object.priceUpdate !== undefined && object.priceUpdate !== null ? exports.MarketPriceUpdateEventV1.fromPartial(object.priceUpdate) : undefined;
        message.marketCreate = object.marketCreate !== undefined && object.marketCreate !== null ? exports.MarketCreateEventV1.fromPartial(object.marketCreate) : undefined;
        message.marketModify = object.marketModify !== undefined && object.marketModify !== null ? exports.MarketModifyEventV1.fromPartial(object.marketModify) : undefined;
        return message;
    }
};
function createBaseMarketPriceUpdateEventV1() {
    return {
        priceWithExponent: helpers_1.Long.UZERO
    };
}
exports.MarketPriceUpdateEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (!message.priceWithExponent.isZero()) {
            writer.uint32(8).uint64(message.priceWithExponent);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMarketPriceUpdateEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.priceWithExponent = reader.uint64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseMarketPriceUpdateEventV1();
        message.priceWithExponent = object.priceWithExponent !== undefined && object.priceWithExponent !== null ? helpers_1.Long.fromValue(object.priceWithExponent) : helpers_1.Long.UZERO;
        return message;
    }
};
function createBaseMarketBaseEventV1() {
    return {
        pair: "",
        minPriceChangePpm: 0
    };
}
exports.MarketBaseEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.pair !== "") {
            writer.uint32(10).string(message.pair);
        }
        if (message.minPriceChangePpm !== 0) {
            writer.uint32(16).uint32(message.minPriceChangePpm);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMarketBaseEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.pair = reader.string();
                    break;
                case 2:
                    message.minPriceChangePpm = reader.uint32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseMarketBaseEventV1();
        message.pair = (_a = object.pair) !== null && _a !== void 0 ? _a : "";
        message.minPriceChangePpm = (_b = object.minPriceChangePpm) !== null && _b !== void 0 ? _b : 0;
        return message;
    }
};
function createBaseMarketCreateEventV1() {
    return {
        base: undefined,
        exponent: 0
    };
}
exports.MarketCreateEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.base !== undefined) {
            exports.MarketBaseEventV1.encode(message.base, writer.uint32(10).fork()).ldelim();
        }
        if (message.exponent !== 0) {
            writer.uint32(16).sint32(message.exponent);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMarketCreateEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.base = exports.MarketBaseEventV1.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.exponent = reader.sint32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseMarketCreateEventV1();
        message.base = object.base !== undefined && object.base !== null ? exports.MarketBaseEventV1.fromPartial(object.base) : undefined;
        message.exponent = (_a = object.exponent) !== null && _a !== void 0 ? _a : 0;
        return message;
    }
};
function createBaseMarketModifyEventV1() {
    return {
        base: undefined
    };
}
exports.MarketModifyEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.base !== undefined) {
            exports.MarketBaseEventV1.encode(message.base, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMarketModifyEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.base = exports.MarketBaseEventV1.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseMarketModifyEventV1();
        message.base = object.base !== undefined && object.base !== null ? exports.MarketBaseEventV1.fromPartial(object.base) : undefined;
        return message;
    }
};
function createBaseSourceOfFunds() {
    return {
        subaccountId: undefined,
        address: undefined
    };
}
exports.SourceOfFunds = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.subaccountId !== undefined) {
            subaccount_1.IndexerSubaccountId.encode(message.subaccountId, writer.uint32(10).fork()).ldelim();
        }
        if (message.address !== undefined) {
            writer.uint32(18).string(message.address);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSourceOfFunds();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.subaccountId = subaccount_1.IndexerSubaccountId.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.address = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseSourceOfFunds();
        message.subaccountId = object.subaccountId !== undefined && object.subaccountId !== null ? subaccount_1.IndexerSubaccountId.fromPartial(object.subaccountId) : undefined;
        message.address = (_a = object.address) !== null && _a !== void 0 ? _a : undefined;
        return message;
    }
};
function createBaseTransferEventV1() {
    return {
        senderSubaccountId: undefined,
        recipientSubaccountId: undefined,
        assetId: 0,
        amount: helpers_1.Long.UZERO,
        sender: undefined,
        recipient: undefined
    };
}
exports.TransferEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.senderSubaccountId !== undefined) {
            subaccount_1.IndexerSubaccountId.encode(message.senderSubaccountId, writer.uint32(10).fork()).ldelim();
        }
        if (message.recipientSubaccountId !== undefined) {
            subaccount_1.IndexerSubaccountId.encode(message.recipientSubaccountId, writer.uint32(18).fork()).ldelim();
        }
        if (message.assetId !== 0) {
            writer.uint32(24).uint32(message.assetId);
        }
        if (!message.amount.isZero()) {
            writer.uint32(32).uint64(message.amount);
        }
        if (message.sender !== undefined) {
            exports.SourceOfFunds.encode(message.sender, writer.uint32(42).fork()).ldelim();
        }
        if (message.recipient !== undefined) {
            exports.SourceOfFunds.encode(message.recipient, writer.uint32(50).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseTransferEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.senderSubaccountId = subaccount_1.IndexerSubaccountId.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.recipientSubaccountId = subaccount_1.IndexerSubaccountId.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.assetId = reader.uint32();
                    break;
                case 4:
                    message.amount = reader.uint64();
                    break;
                case 5:
                    message.sender = exports.SourceOfFunds.decode(reader, reader.uint32());
                    break;
                case 6:
                    message.recipient = exports.SourceOfFunds.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseTransferEventV1();
        message.senderSubaccountId = object.senderSubaccountId !== undefined && object.senderSubaccountId !== null ? subaccount_1.IndexerSubaccountId.fromPartial(object.senderSubaccountId) : undefined;
        message.recipientSubaccountId = object.recipientSubaccountId !== undefined && object.recipientSubaccountId !== null ? subaccount_1.IndexerSubaccountId.fromPartial(object.recipientSubaccountId) : undefined;
        message.assetId = (_a = object.assetId) !== null && _a !== void 0 ? _a : 0;
        message.amount = object.amount !== undefined && object.amount !== null ? helpers_1.Long.fromValue(object.amount) : helpers_1.Long.UZERO;
        message.sender = object.sender !== undefined && object.sender !== null ? exports.SourceOfFunds.fromPartial(object.sender) : undefined;
        message.recipient = object.recipient !== undefined && object.recipient !== null ? exports.SourceOfFunds.fromPartial(object.recipient) : undefined;
        return message;
    }
};
function createBaseOrderFillEventV1() {
    return {
        makerOrder: undefined,
        order: undefined,
        liquidationOrder: undefined,
        fillAmount: helpers_1.Long.UZERO,
        makerFee: helpers_1.Long.ZERO,
        takerFee: helpers_1.Long.ZERO,
        totalFilledMaker: helpers_1.Long.UZERO,
        totalFilledTaker: helpers_1.Long.UZERO,
        affiliateRevShare: helpers_1.Long.UZERO
    };
}
exports.OrderFillEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.makerOrder !== undefined) {
            clob_1.IndexerOrder.encode(message.makerOrder, writer.uint32(10).fork()).ldelim();
        }
        if (message.order !== undefined) {
            clob_1.IndexerOrder.encode(message.order, writer.uint32(18).fork()).ldelim();
        }
        if (message.liquidationOrder !== undefined) {
            exports.LiquidationOrderV1.encode(message.liquidationOrder, writer.uint32(34).fork()).ldelim();
        }
        if (!message.fillAmount.isZero()) {
            writer.uint32(24).uint64(message.fillAmount);
        }
        if (!message.makerFee.isZero()) {
            writer.uint32(40).sint64(message.makerFee);
        }
        if (!message.takerFee.isZero()) {
            writer.uint32(48).sint64(message.takerFee);
        }
        if (!message.totalFilledMaker.isZero()) {
            writer.uint32(56).uint64(message.totalFilledMaker);
        }
        if (!message.totalFilledTaker.isZero()) {
            writer.uint32(64).uint64(message.totalFilledTaker);
        }
        if (!message.affiliateRevShare.isZero()) {
            writer.uint32(72).uint64(message.affiliateRevShare);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseOrderFillEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.makerOrder = clob_1.IndexerOrder.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.order = clob_1.IndexerOrder.decode(reader, reader.uint32());
                    break;
                case 4:
                    message.liquidationOrder = exports.LiquidationOrderV1.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.fillAmount = reader.uint64();
                    break;
                case 5:
                    message.makerFee = reader.sint64();
                    break;
                case 6:
                    message.takerFee = reader.sint64();
                    break;
                case 7:
                    message.totalFilledMaker = reader.uint64();
                    break;
                case 8:
                    message.totalFilledTaker = reader.uint64();
                    break;
                case 9:
                    message.affiliateRevShare = reader.uint64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseOrderFillEventV1();
        message.makerOrder = object.makerOrder !== undefined && object.makerOrder !== null ? clob_1.IndexerOrder.fromPartial(object.makerOrder) : undefined;
        message.order = object.order !== undefined && object.order !== null ? clob_1.IndexerOrder.fromPartial(object.order) : undefined;
        message.liquidationOrder = object.liquidationOrder !== undefined && object.liquidationOrder !== null ? exports.LiquidationOrderV1.fromPartial(object.liquidationOrder) : undefined;
        message.fillAmount = object.fillAmount !== undefined && object.fillAmount !== null ? helpers_1.Long.fromValue(object.fillAmount) : helpers_1.Long.UZERO;
        message.makerFee = object.makerFee !== undefined && object.makerFee !== null ? helpers_1.Long.fromValue(object.makerFee) : helpers_1.Long.ZERO;
        message.takerFee = object.takerFee !== undefined && object.takerFee !== null ? helpers_1.Long.fromValue(object.takerFee) : helpers_1.Long.ZERO;
        message.totalFilledMaker = object.totalFilledMaker !== undefined && object.totalFilledMaker !== null ? helpers_1.Long.fromValue(object.totalFilledMaker) : helpers_1.Long.UZERO;
        message.totalFilledTaker = object.totalFilledTaker !== undefined && object.totalFilledTaker !== null ? helpers_1.Long.fromValue(object.totalFilledTaker) : helpers_1.Long.UZERO;
        message.affiliateRevShare = object.affiliateRevShare !== undefined && object.affiliateRevShare !== null ? helpers_1.Long.fromValue(object.affiliateRevShare) : helpers_1.Long.UZERO;
        return message;
    }
};
function createBaseDeleveragingEventV1() {
    return {
        liquidated: undefined,
        offsetting: undefined,
        perpetualId: 0,
        fillAmount: helpers_1.Long.UZERO,
        totalQuoteQuantums: helpers_1.Long.UZERO,
        isBuy: false,
        isFinalSettlement: false
    };
}
exports.DeleveragingEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.liquidated !== undefined) {
            subaccount_1.IndexerSubaccountId.encode(message.liquidated, writer.uint32(10).fork()).ldelim();
        }
        if (message.offsetting !== undefined) {
            subaccount_1.IndexerSubaccountId.encode(message.offsetting, writer.uint32(18).fork()).ldelim();
        }
        if (message.perpetualId !== 0) {
            writer.uint32(24).uint32(message.perpetualId);
        }
        if (!message.fillAmount.isZero()) {
            writer.uint32(32).uint64(message.fillAmount);
        }
        if (!message.totalQuoteQuantums.isZero()) {
            writer.uint32(40).uint64(message.totalQuoteQuantums);
        }
        if (message.isBuy === true) {
            writer.uint32(48).bool(message.isBuy);
        }
        if (message.isFinalSettlement === true) {
            writer.uint32(56).bool(message.isFinalSettlement);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDeleveragingEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.liquidated = subaccount_1.IndexerSubaccountId.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.offsetting = subaccount_1.IndexerSubaccountId.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.perpetualId = reader.uint32();
                    break;
                case 4:
                    message.fillAmount = reader.uint64();
                    break;
                case 5:
                    message.totalQuoteQuantums = reader.uint64();
                    break;
                case 6:
                    message.isBuy = reader.bool();
                    break;
                case 7:
                    message.isFinalSettlement = reader.bool();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseDeleveragingEventV1();
        message.liquidated = object.liquidated !== undefined && object.liquidated !== null ? subaccount_1.IndexerSubaccountId.fromPartial(object.liquidated) : undefined;
        message.offsetting = object.offsetting !== undefined && object.offsetting !== null ? subaccount_1.IndexerSubaccountId.fromPartial(object.offsetting) : undefined;
        message.perpetualId = (_a = object.perpetualId) !== null && _a !== void 0 ? _a : 0;
        message.fillAmount = object.fillAmount !== undefined && object.fillAmount !== null ? helpers_1.Long.fromValue(object.fillAmount) : helpers_1.Long.UZERO;
        message.totalQuoteQuantums = object.totalQuoteQuantums !== undefined && object.totalQuoteQuantums !== null ? helpers_1.Long.fromValue(object.totalQuoteQuantums) : helpers_1.Long.UZERO;
        message.isBuy = (_b = object.isBuy) !== null && _b !== void 0 ? _b : false;
        message.isFinalSettlement = (_c = object.isFinalSettlement) !== null && _c !== void 0 ? _c : false;
        return message;
    }
};
function createBaseLiquidationOrderV1() {
    return {
        liquidated: undefined,
        clobPairId: 0,
        perpetualId: 0,
        totalSize: helpers_1.Long.UZERO,
        isBuy: false,
        subticks: helpers_1.Long.UZERO
    };
}
exports.LiquidationOrderV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.liquidated !== undefined) {
            subaccount_1.IndexerSubaccountId.encode(message.liquidated, writer.uint32(10).fork()).ldelim();
        }
        if (message.clobPairId !== 0) {
            writer.uint32(16).uint32(message.clobPairId);
        }
        if (message.perpetualId !== 0) {
            writer.uint32(24).uint32(message.perpetualId);
        }
        if (!message.totalSize.isZero()) {
            writer.uint32(32).uint64(message.totalSize);
        }
        if (message.isBuy === true) {
            writer.uint32(40).bool(message.isBuy);
        }
        if (!message.subticks.isZero()) {
            writer.uint32(48).uint64(message.subticks);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseLiquidationOrderV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.liquidated = subaccount_1.IndexerSubaccountId.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.clobPairId = reader.uint32();
                    break;
                case 3:
                    message.perpetualId = reader.uint32();
                    break;
                case 4:
                    message.totalSize = reader.uint64();
                    break;
                case 5:
                    message.isBuy = reader.bool();
                    break;
                case 6:
                    message.subticks = reader.uint64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseLiquidationOrderV1();
        message.liquidated = object.liquidated !== undefined && object.liquidated !== null ? subaccount_1.IndexerSubaccountId.fromPartial(object.liquidated) : undefined;
        message.clobPairId = (_a = object.clobPairId) !== null && _a !== void 0 ? _a : 0;
        message.perpetualId = (_b = object.perpetualId) !== null && _b !== void 0 ? _b : 0;
        message.totalSize = object.totalSize !== undefined && object.totalSize !== null ? helpers_1.Long.fromValue(object.totalSize) : helpers_1.Long.UZERO;
        message.isBuy = (_c = object.isBuy) !== null && _c !== void 0 ? _c : false;
        message.subticks = object.subticks !== undefined && object.subticks !== null ? helpers_1.Long.fromValue(object.subticks) : helpers_1.Long.UZERO;
        return message;
    }
};
function createBaseSubaccountUpdateEventV1() {
    return {
        subaccountId: undefined,
        updatedPerpetualPositions: [],
        updatedAssetPositions: []
    };
}
exports.SubaccountUpdateEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.subaccountId !== undefined) {
            subaccount_1.IndexerSubaccountId.encode(message.subaccountId, writer.uint32(10).fork()).ldelim();
        }
        for (const v of message.updatedPerpetualPositions) {
            subaccount_1.IndexerPerpetualPosition.encode(v, writer.uint32(26).fork()).ldelim();
        }
        for (const v of message.updatedAssetPositions) {
            subaccount_1.IndexerAssetPosition.encode(v, writer.uint32(34).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSubaccountUpdateEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.subaccountId = subaccount_1.IndexerSubaccountId.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.updatedPerpetualPositions.push(subaccount_1.IndexerPerpetualPosition.decode(reader, reader.uint32()));
                    break;
                case 4:
                    message.updatedAssetPositions.push(subaccount_1.IndexerAssetPosition.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseSubaccountUpdateEventV1();
        message.subaccountId = object.subaccountId !== undefined && object.subaccountId !== null ? subaccount_1.IndexerSubaccountId.fromPartial(object.subaccountId) : undefined;
        message.updatedPerpetualPositions = ((_a = object.updatedPerpetualPositions) === null || _a === void 0 ? void 0 : _a.map(e => subaccount_1.IndexerPerpetualPosition.fromPartial(e))) || [];
        message.updatedAssetPositions = ((_b = object.updatedAssetPositions) === null || _b === void 0 ? void 0 : _b.map(e => subaccount_1.IndexerAssetPosition.fromPartial(e))) || [];
        return message;
    }
};
function createBaseStatefulOrderEventV1() {
    return {
        orderPlace: undefined,
        orderRemoval: undefined,
        conditionalOrderPlacement: undefined,
        conditionalOrderTriggered: undefined,
        longTermOrderPlacement: undefined,
        orderReplacement: undefined
    };
}
exports.StatefulOrderEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.orderPlace !== undefined) {
            exports.StatefulOrderEventV1_StatefulOrderPlacementV1.encode(message.orderPlace, writer.uint32(10).fork()).ldelim();
        }
        if (message.orderRemoval !== undefined) {
            exports.StatefulOrderEventV1_StatefulOrderRemovalV1.encode(message.orderRemoval, writer.uint32(34).fork()).ldelim();
        }
        if (message.conditionalOrderPlacement !== undefined) {
            exports.StatefulOrderEventV1_ConditionalOrderPlacementV1.encode(message.conditionalOrderPlacement, writer.uint32(42).fork()).ldelim();
        }
        if (message.conditionalOrderTriggered !== undefined) {
            exports.StatefulOrderEventV1_ConditionalOrderTriggeredV1.encode(message.conditionalOrderTriggered, writer.uint32(50).fork()).ldelim();
        }
        if (message.longTermOrderPlacement !== undefined) {
            exports.StatefulOrderEventV1_LongTermOrderPlacementV1.encode(message.longTermOrderPlacement, writer.uint32(58).fork()).ldelim();
        }
        if (message.orderReplacement !== undefined) {
            exports.StatefulOrderEventV1_LongTermOrderReplacementV1.encode(message.orderReplacement, writer.uint32(66).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseStatefulOrderEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.orderPlace = exports.StatefulOrderEventV1_StatefulOrderPlacementV1.decode(reader, reader.uint32());
                    break;
                case 4:
                    message.orderRemoval = exports.StatefulOrderEventV1_StatefulOrderRemovalV1.decode(reader, reader.uint32());
                    break;
                case 5:
                    message.conditionalOrderPlacement = exports.StatefulOrderEventV1_ConditionalOrderPlacementV1.decode(reader, reader.uint32());
                    break;
                case 6:
                    message.conditionalOrderTriggered = exports.StatefulOrderEventV1_ConditionalOrderTriggeredV1.decode(reader, reader.uint32());
                    break;
                case 7:
                    message.longTermOrderPlacement = exports.StatefulOrderEventV1_LongTermOrderPlacementV1.decode(reader, reader.uint32());
                    break;
                case 8:
                    message.orderReplacement = exports.StatefulOrderEventV1_LongTermOrderReplacementV1.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseStatefulOrderEventV1();
        message.orderPlace = object.orderPlace !== undefined && object.orderPlace !== null ? exports.StatefulOrderEventV1_StatefulOrderPlacementV1.fromPartial(object.orderPlace) : undefined;
        message.orderRemoval = object.orderRemoval !== undefined && object.orderRemoval !== null ? exports.StatefulOrderEventV1_StatefulOrderRemovalV1.fromPartial(object.orderRemoval) : undefined;
        message.conditionalOrderPlacement = object.conditionalOrderPlacement !== undefined && object.conditionalOrderPlacement !== null ? exports.StatefulOrderEventV1_ConditionalOrderPlacementV1.fromPartial(object.conditionalOrderPlacement) : undefined;
        message.conditionalOrderTriggered = object.conditionalOrderTriggered !== undefined && object.conditionalOrderTriggered !== null ? exports.StatefulOrderEventV1_ConditionalOrderTriggeredV1.fromPartial(object.conditionalOrderTriggered) : undefined;
        message.longTermOrderPlacement = object.longTermOrderPlacement !== undefined && object.longTermOrderPlacement !== null ? exports.StatefulOrderEventV1_LongTermOrderPlacementV1.fromPartial(object.longTermOrderPlacement) : undefined;
        message.orderReplacement = object.orderReplacement !== undefined && object.orderReplacement !== null ? exports.StatefulOrderEventV1_LongTermOrderReplacementV1.fromPartial(object.orderReplacement) : undefined;
        return message;
    }
};
function createBaseStatefulOrderEventV1_StatefulOrderPlacementV1() {
    return {
        order: undefined
    };
}
exports.StatefulOrderEventV1_StatefulOrderPlacementV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.order !== undefined) {
            clob_1.IndexerOrder.encode(message.order, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseStatefulOrderEventV1_StatefulOrderPlacementV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.order = clob_1.IndexerOrder.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseStatefulOrderEventV1_StatefulOrderPlacementV1();
        message.order = object.order !== undefined && object.order !== null ? clob_1.IndexerOrder.fromPartial(object.order) : undefined;
        return message;
    }
};
function createBaseStatefulOrderEventV1_StatefulOrderRemovalV1() {
    return {
        removedOrderId: undefined,
        reason: 0
    };
}
exports.StatefulOrderEventV1_StatefulOrderRemovalV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.removedOrderId !== undefined) {
            clob_1.IndexerOrderId.encode(message.removedOrderId, writer.uint32(10).fork()).ldelim();
        }
        if (message.reason !== 0) {
            writer.uint32(16).int32(message.reason);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseStatefulOrderEventV1_StatefulOrderRemovalV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.removedOrderId = clob_1.IndexerOrderId.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.reason = reader.int32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseStatefulOrderEventV1_StatefulOrderRemovalV1();
        message.removedOrderId = object.removedOrderId !== undefined && object.removedOrderId !== null ? clob_1.IndexerOrderId.fromPartial(object.removedOrderId) : undefined;
        message.reason = (_a = object.reason) !== null && _a !== void 0 ? _a : 0;
        return message;
    }
};
function createBaseStatefulOrderEventV1_ConditionalOrderPlacementV1() {
    return {
        order: undefined
    };
}
exports.StatefulOrderEventV1_ConditionalOrderPlacementV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.order !== undefined) {
            clob_1.IndexerOrder.encode(message.order, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseStatefulOrderEventV1_ConditionalOrderPlacementV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.order = clob_1.IndexerOrder.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseStatefulOrderEventV1_ConditionalOrderPlacementV1();
        message.order = object.order !== undefined && object.order !== null ? clob_1.IndexerOrder.fromPartial(object.order) : undefined;
        return message;
    }
};
function createBaseStatefulOrderEventV1_ConditionalOrderTriggeredV1() {
    return {
        triggeredOrderId: undefined
    };
}
exports.StatefulOrderEventV1_ConditionalOrderTriggeredV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.triggeredOrderId !== undefined) {
            clob_1.IndexerOrderId.encode(message.triggeredOrderId, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseStatefulOrderEventV1_ConditionalOrderTriggeredV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.triggeredOrderId = clob_1.IndexerOrderId.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseStatefulOrderEventV1_ConditionalOrderTriggeredV1();
        message.triggeredOrderId = object.triggeredOrderId !== undefined && object.triggeredOrderId !== null ? clob_1.IndexerOrderId.fromPartial(object.triggeredOrderId) : undefined;
        return message;
    }
};
function createBaseStatefulOrderEventV1_LongTermOrderPlacementV1() {
    return {
        order: undefined
    };
}
exports.StatefulOrderEventV1_LongTermOrderPlacementV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.order !== undefined) {
            clob_1.IndexerOrder.encode(message.order, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseStatefulOrderEventV1_LongTermOrderPlacementV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.order = clob_1.IndexerOrder.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseStatefulOrderEventV1_LongTermOrderPlacementV1();
        message.order = object.order !== undefined && object.order !== null ? clob_1.IndexerOrder.fromPartial(object.order) : undefined;
        return message;
    }
};
function createBaseStatefulOrderEventV1_LongTermOrderReplacementV1() {
    return {
        oldOrderId: undefined,
        order: undefined
    };
}
exports.StatefulOrderEventV1_LongTermOrderReplacementV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.oldOrderId !== undefined) {
            clob_1.IndexerOrderId.encode(message.oldOrderId, writer.uint32(10).fork()).ldelim();
        }
        if (message.order !== undefined) {
            clob_1.IndexerOrder.encode(message.order, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseStatefulOrderEventV1_LongTermOrderReplacementV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.oldOrderId = clob_1.IndexerOrderId.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.order = clob_1.IndexerOrder.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseStatefulOrderEventV1_LongTermOrderReplacementV1();
        message.oldOrderId = object.oldOrderId !== undefined && object.oldOrderId !== null ? clob_1.IndexerOrderId.fromPartial(object.oldOrderId) : undefined;
        message.order = object.order !== undefined && object.order !== null ? clob_1.IndexerOrder.fromPartial(object.order) : undefined;
        return message;
    }
};
function createBaseAssetCreateEventV1() {
    return {
        id: 0,
        symbol: "",
        hasMarket: false,
        marketId: 0,
        atomicResolution: 0
    };
}
exports.AssetCreateEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.id !== 0) {
            writer.uint32(8).uint32(message.id);
        }
        if (message.symbol !== "") {
            writer.uint32(18).string(message.symbol);
        }
        if (message.hasMarket === true) {
            writer.uint32(24).bool(message.hasMarket);
        }
        if (message.marketId !== 0) {
            writer.uint32(32).uint32(message.marketId);
        }
        if (message.atomicResolution !== 0) {
            writer.uint32(40).sint32(message.atomicResolution);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAssetCreateEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.id = reader.uint32();
                    break;
                case 2:
                    message.symbol = reader.string();
                    break;
                case 3:
                    message.hasMarket = reader.bool();
                    break;
                case 4:
                    message.marketId = reader.uint32();
                    break;
                case 5:
                    message.atomicResolution = reader.sint32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e;
        const message = createBaseAssetCreateEventV1();
        message.id = (_a = object.id) !== null && _a !== void 0 ? _a : 0;
        message.symbol = (_b = object.symbol) !== null && _b !== void 0 ? _b : "";
        message.hasMarket = (_c = object.hasMarket) !== null && _c !== void 0 ? _c : false;
        message.marketId = (_d = object.marketId) !== null && _d !== void 0 ? _d : 0;
        message.atomicResolution = (_e = object.atomicResolution) !== null && _e !== void 0 ? _e : 0;
        return message;
    }
};
function createBasePerpetualMarketCreateEventV1() {
    return {
        id: 0,
        clobPairId: 0,
        ticker: "",
        marketId: 0,
        status: 0,
        quantumConversionExponent: 0,
        atomicResolution: 0,
        subticksPerTick: 0,
        stepBaseQuantums: helpers_1.Long.UZERO,
        liquidityTier: 0
    };
}
exports.PerpetualMarketCreateEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.id !== 0) {
            writer.uint32(8).uint32(message.id);
        }
        if (message.clobPairId !== 0) {
            writer.uint32(16).uint32(message.clobPairId);
        }
        if (message.ticker !== "") {
            writer.uint32(26).string(message.ticker);
        }
        if (message.marketId !== 0) {
            writer.uint32(32).uint32(message.marketId);
        }
        if (message.status !== 0) {
            writer.uint32(40).int32(message.status);
        }
        if (message.quantumConversionExponent !== 0) {
            writer.uint32(48).sint32(message.quantumConversionExponent);
        }
        if (message.atomicResolution !== 0) {
            writer.uint32(56).sint32(message.atomicResolution);
        }
        if (message.subticksPerTick !== 0) {
            writer.uint32(64).uint32(message.subticksPerTick);
        }
        if (!message.stepBaseQuantums.isZero()) {
            writer.uint32(72).uint64(message.stepBaseQuantums);
        }
        if (message.liquidityTier !== 0) {
            writer.uint32(80).uint32(message.liquidityTier);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePerpetualMarketCreateEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.id = reader.uint32();
                    break;
                case 2:
                    message.clobPairId = reader.uint32();
                    break;
                case 3:
                    message.ticker = reader.string();
                    break;
                case 4:
                    message.marketId = reader.uint32();
                    break;
                case 5:
                    message.status = reader.int32();
                    break;
                case 6:
                    message.quantumConversionExponent = reader.sint32();
                    break;
                case 7:
                    message.atomicResolution = reader.sint32();
                    break;
                case 8:
                    message.subticksPerTick = reader.uint32();
                    break;
                case 9:
                    message.stepBaseQuantums = reader.uint64();
                    break;
                case 10:
                    message.liquidityTier = reader.uint32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j;
        const message = createBasePerpetualMarketCreateEventV1();
        message.id = (_a = object.id) !== null && _a !== void 0 ? _a : 0;
        message.clobPairId = (_b = object.clobPairId) !== null && _b !== void 0 ? _b : 0;
        message.ticker = (_c = object.ticker) !== null && _c !== void 0 ? _c : "";
        message.marketId = (_d = object.marketId) !== null && _d !== void 0 ? _d : 0;
        message.status = (_e = object.status) !== null && _e !== void 0 ? _e : 0;
        message.quantumConversionExponent = (_f = object.quantumConversionExponent) !== null && _f !== void 0 ? _f : 0;
        message.atomicResolution = (_g = object.atomicResolution) !== null && _g !== void 0 ? _g : 0;
        message.subticksPerTick = (_h = object.subticksPerTick) !== null && _h !== void 0 ? _h : 0;
        message.stepBaseQuantums = object.stepBaseQuantums !== undefined && object.stepBaseQuantums !== null ? helpers_1.Long.fromValue(object.stepBaseQuantums) : helpers_1.Long.UZERO;
        message.liquidityTier = (_j = object.liquidityTier) !== null && _j !== void 0 ? _j : 0;
        return message;
    }
};
function createBasePerpetualMarketCreateEventV2() {
    return {
        id: 0,
        clobPairId: 0,
        ticker: "",
        marketId: 0,
        status: 0,
        quantumConversionExponent: 0,
        atomicResolution: 0,
        subticksPerTick: 0,
        stepBaseQuantums: helpers_1.Long.UZERO,
        liquidityTier: 0,
        marketType: 0
    };
}
exports.PerpetualMarketCreateEventV2 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.id !== 0) {
            writer.uint32(8).uint32(message.id);
        }
        if (message.clobPairId !== 0) {
            writer.uint32(16).uint32(message.clobPairId);
        }
        if (message.ticker !== "") {
            writer.uint32(26).string(message.ticker);
        }
        if (message.marketId !== 0) {
            writer.uint32(32).uint32(message.marketId);
        }
        if (message.status !== 0) {
            writer.uint32(40).int32(message.status);
        }
        if (message.quantumConversionExponent !== 0) {
            writer.uint32(48).sint32(message.quantumConversionExponent);
        }
        if (message.atomicResolution !== 0) {
            writer.uint32(56).sint32(message.atomicResolution);
        }
        if (message.subticksPerTick !== 0) {
            writer.uint32(64).uint32(message.subticksPerTick);
        }
        if (!message.stepBaseQuantums.isZero()) {
            writer.uint32(72).uint64(message.stepBaseQuantums);
        }
        if (message.liquidityTier !== 0) {
            writer.uint32(80).uint32(message.liquidityTier);
        }
        if (message.marketType !== 0) {
            writer.uint32(88).int32(message.marketType);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePerpetualMarketCreateEventV2();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.id = reader.uint32();
                    break;
                case 2:
                    message.clobPairId = reader.uint32();
                    break;
                case 3:
                    message.ticker = reader.string();
                    break;
                case 4:
                    message.marketId = reader.uint32();
                    break;
                case 5:
                    message.status = reader.int32();
                    break;
                case 6:
                    message.quantumConversionExponent = reader.sint32();
                    break;
                case 7:
                    message.atomicResolution = reader.sint32();
                    break;
                case 8:
                    message.subticksPerTick = reader.uint32();
                    break;
                case 9:
                    message.stepBaseQuantums = reader.uint64();
                    break;
                case 10:
                    message.liquidityTier = reader.uint32();
                    break;
                case 11:
                    message.marketType = reader.int32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k;
        const message = createBasePerpetualMarketCreateEventV2();
        message.id = (_a = object.id) !== null && _a !== void 0 ? _a : 0;
        message.clobPairId = (_b = object.clobPairId) !== null && _b !== void 0 ? _b : 0;
        message.ticker = (_c = object.ticker) !== null && _c !== void 0 ? _c : "";
        message.marketId = (_d = object.marketId) !== null && _d !== void 0 ? _d : 0;
        message.status = (_e = object.status) !== null && _e !== void 0 ? _e : 0;
        message.quantumConversionExponent = (_f = object.quantumConversionExponent) !== null && _f !== void 0 ? _f : 0;
        message.atomicResolution = (_g = object.atomicResolution) !== null && _g !== void 0 ? _g : 0;
        message.subticksPerTick = (_h = object.subticksPerTick) !== null && _h !== void 0 ? _h : 0;
        message.stepBaseQuantums = object.stepBaseQuantums !== undefined && object.stepBaseQuantums !== null ? helpers_1.Long.fromValue(object.stepBaseQuantums) : helpers_1.Long.UZERO;
        message.liquidityTier = (_j = object.liquidityTier) !== null && _j !== void 0 ? _j : 0;
        message.marketType = (_k = object.marketType) !== null && _k !== void 0 ? _k : 0;
        return message;
    }
};
function createBaseLiquidityTierUpsertEventV1() {
    return {
        id: 0,
        name: "",
        initialMarginPpm: 0,
        maintenanceFractionPpm: 0,
        basePositionNotional: helpers_1.Long.UZERO
    };
}
exports.LiquidityTierUpsertEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.id !== 0) {
            writer.uint32(8).uint32(message.id);
        }
        if (message.name !== "") {
            writer.uint32(18).string(message.name);
        }
        if (message.initialMarginPpm !== 0) {
            writer.uint32(24).uint32(message.initialMarginPpm);
        }
        if (message.maintenanceFractionPpm !== 0) {
            writer.uint32(32).uint32(message.maintenanceFractionPpm);
        }
        if (!message.basePositionNotional.isZero()) {
            writer.uint32(40).uint64(message.basePositionNotional);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseLiquidityTierUpsertEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.id = reader.uint32();
                    break;
                case 2:
                    message.name = reader.string();
                    break;
                case 3:
                    message.initialMarginPpm = reader.uint32();
                    break;
                case 4:
                    message.maintenanceFractionPpm = reader.uint32();
                    break;
                case 5:
                    message.basePositionNotional = reader.uint64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c, _d;
        const message = createBaseLiquidityTierUpsertEventV1();
        message.id = (_a = object.id) !== null && _a !== void 0 ? _a : 0;
        message.name = (_b = object.name) !== null && _b !== void 0 ? _b : "";
        message.initialMarginPpm = (_c = object.initialMarginPpm) !== null && _c !== void 0 ? _c : 0;
        message.maintenanceFractionPpm = (_d = object.maintenanceFractionPpm) !== null && _d !== void 0 ? _d : 0;
        message.basePositionNotional = object.basePositionNotional !== undefined && object.basePositionNotional !== null ? helpers_1.Long.fromValue(object.basePositionNotional) : helpers_1.Long.UZERO;
        return message;
    }
};
function createBaseUpdateClobPairEventV1() {
    return {
        clobPairId: 0,
        status: 0,
        quantumConversionExponent: 0,
        subticksPerTick: 0,
        stepBaseQuantums: helpers_1.Long.UZERO
    };
}
exports.UpdateClobPairEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.clobPairId !== 0) {
            writer.uint32(8).uint32(message.clobPairId);
        }
        if (message.status !== 0) {
            writer.uint32(16).int32(message.status);
        }
        if (message.quantumConversionExponent !== 0) {
            writer.uint32(24).sint32(message.quantumConversionExponent);
        }
        if (message.subticksPerTick !== 0) {
            writer.uint32(32).uint32(message.subticksPerTick);
        }
        if (!message.stepBaseQuantums.isZero()) {
            writer.uint32(40).uint64(message.stepBaseQuantums);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateClobPairEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.clobPairId = reader.uint32();
                    break;
                case 2:
                    message.status = reader.int32();
                    break;
                case 3:
                    message.quantumConversionExponent = reader.sint32();
                    break;
                case 4:
                    message.subticksPerTick = reader.uint32();
                    break;
                case 5:
                    message.stepBaseQuantums = reader.uint64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c, _d;
        const message = createBaseUpdateClobPairEventV1();
        message.clobPairId = (_a = object.clobPairId) !== null && _a !== void 0 ? _a : 0;
        message.status = (_b = object.status) !== null && _b !== void 0 ? _b : 0;
        message.quantumConversionExponent = (_c = object.quantumConversionExponent) !== null && _c !== void 0 ? _c : 0;
        message.subticksPerTick = (_d = object.subticksPerTick) !== null && _d !== void 0 ? _d : 0;
        message.stepBaseQuantums = object.stepBaseQuantums !== undefined && object.stepBaseQuantums !== null ? helpers_1.Long.fromValue(object.stepBaseQuantums) : helpers_1.Long.UZERO;
        return message;
    }
};
function createBaseUpdatePerpetualEventV1() {
    return {
        id: 0,
        ticker: "",
        marketId: 0,
        atomicResolution: 0,
        liquidityTier: 0
    };
}
exports.UpdatePerpetualEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.id !== 0) {
            writer.uint32(8).uint32(message.id);
        }
        if (message.ticker !== "") {
            writer.uint32(18).string(message.ticker);
        }
        if (message.marketId !== 0) {
            writer.uint32(24).uint32(message.marketId);
        }
        if (message.atomicResolution !== 0) {
            writer.uint32(32).sint32(message.atomicResolution);
        }
        if (message.liquidityTier !== 0) {
            writer.uint32(40).uint32(message.liquidityTier);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdatePerpetualEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.id = reader.uint32();
                    break;
                case 2:
                    message.ticker = reader.string();
                    break;
                case 3:
                    message.marketId = reader.uint32();
                    break;
                case 4:
                    message.atomicResolution = reader.sint32();
                    break;
                case 5:
                    message.liquidityTier = reader.uint32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e;
        const message = createBaseUpdatePerpetualEventV1();
        message.id = (_a = object.id) !== null && _a !== void 0 ? _a : 0;
        message.ticker = (_b = object.ticker) !== null && _b !== void 0 ? _b : "";
        message.marketId = (_c = object.marketId) !== null && _c !== void 0 ? _c : 0;
        message.atomicResolution = (_d = object.atomicResolution) !== null && _d !== void 0 ? _d : 0;
        message.liquidityTier = (_e = object.liquidityTier) !== null && _e !== void 0 ? _e : 0;
        return message;
    }
};
function createBaseUpdatePerpetualEventV2() {
    return {
        id: 0,
        ticker: "",
        marketId: 0,
        atomicResolution: 0,
        liquidityTier: 0,
        marketType: 0
    };
}
exports.UpdatePerpetualEventV2 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.id !== 0) {
            writer.uint32(8).uint32(message.id);
        }
        if (message.ticker !== "") {
            writer.uint32(18).string(message.ticker);
        }
        if (message.marketId !== 0) {
            writer.uint32(24).uint32(message.marketId);
        }
        if (message.atomicResolution !== 0) {
            writer.uint32(32).sint32(message.atomicResolution);
        }
        if (message.liquidityTier !== 0) {
            writer.uint32(40).uint32(message.liquidityTier);
        }
        if (message.marketType !== 0) {
            writer.uint32(48).int32(message.marketType);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdatePerpetualEventV2();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.id = reader.uint32();
                    break;
                case 2:
                    message.ticker = reader.string();
                    break;
                case 3:
                    message.marketId = reader.uint32();
                    break;
                case 4:
                    message.atomicResolution = reader.sint32();
                    break;
                case 5:
                    message.liquidityTier = reader.uint32();
                    break;
                case 6:
                    message.marketType = reader.int32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e, _f;
        const message = createBaseUpdatePerpetualEventV2();
        message.id = (_a = object.id) !== null && _a !== void 0 ? _a : 0;
        message.ticker = (_b = object.ticker) !== null && _b !== void 0 ? _b : "";
        message.marketId = (_c = object.marketId) !== null && _c !== void 0 ? _c : 0;
        message.atomicResolution = (_d = object.atomicResolution) !== null && _d !== void 0 ? _d : 0;
        message.liquidityTier = (_e = object.liquidityTier) !== null && _e !== void 0 ? _e : 0;
        message.marketType = (_f = object.marketType) !== null && _f !== void 0 ? _f : 0;
        return message;
    }
};
function createBaseTradingRewardsEventV1() {
    return {
        tradingRewards: []
    };
}
exports.TradingRewardsEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.tradingRewards) {
            exports.AddressTradingReward.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseTradingRewardsEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.tradingRewards.push(exports.AddressTradingReward.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseTradingRewardsEventV1();
        message.tradingRewards = ((_a = object.tradingRewards) === null || _a === void 0 ? void 0 : _a.map(e => exports.AddressTradingReward.fromPartial(e))) || [];
        return message;
    }
};
function createBaseAddressTradingReward() {
    return {
        owner: "",
        denomAmount: new Uint8Array()
    };
}
exports.AddressTradingReward = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.owner !== "") {
            writer.uint32(10).string(message.owner);
        }
        if (message.denomAmount.length !== 0) {
            writer.uint32(18).bytes(message.denomAmount);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAddressTradingReward();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.owner = reader.string();
                    break;
                case 2:
                    message.denomAmount = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseAddressTradingReward();
        message.owner = (_a = object.owner) !== null && _a !== void 0 ? _a : "";
        message.denomAmount = (_b = object.denomAmount) !== null && _b !== void 0 ? _b : new Uint8Array();
        return message;
    }
};
function createBaseOpenInterestUpdateEventV1() {
    return {
        openInterestUpdates: []
    };
}
exports.OpenInterestUpdateEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.openInterestUpdates) {
            exports.OpenInterestUpdate.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseOpenInterestUpdateEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.openInterestUpdates.push(exports.OpenInterestUpdate.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseOpenInterestUpdateEventV1();
        message.openInterestUpdates = ((_a = object.openInterestUpdates) === null || _a === void 0 ? void 0 : _a.map(e => exports.OpenInterestUpdate.fromPartial(e))) || [];
        return message;
    }
};
function createBaseOpenInterestUpdate() {
    return {
        perpetualId: 0,
        openInterest: new Uint8Array()
    };
}
exports.OpenInterestUpdate = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.perpetualId !== 0) {
            writer.uint32(8).uint32(message.perpetualId);
        }
        if (message.openInterest.length !== 0) {
            writer.uint32(18).bytes(message.openInterest);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseOpenInterestUpdate();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.perpetualId = reader.uint32();
                    break;
                case 2:
                    message.openInterest = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseOpenInterestUpdate();
        message.perpetualId = (_a = object.perpetualId) !== null && _a !== void 0 ? _a : 0;
        message.openInterest = (_b = object.openInterest) !== null && _b !== void 0 ? _b : new Uint8Array();
        return message;
    }
};
function createBaseLiquidityTierUpsertEventV2() {
    return {
        id: 0,
        name: "",
        initialMarginPpm: 0,
        maintenanceFractionPpm: 0,
        basePositionNotional: helpers_1.Long.UZERO,
        openInterestLowerCap: helpers_1.Long.UZERO,
        openInterestUpperCap: helpers_1.Long.UZERO
    };
}
exports.LiquidityTierUpsertEventV2 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.id !== 0) {
            writer.uint32(8).uint32(message.id);
        }
        if (message.name !== "") {
            writer.uint32(18).string(message.name);
        }
        if (message.initialMarginPpm !== 0) {
            writer.uint32(24).uint32(message.initialMarginPpm);
        }
        if (message.maintenanceFractionPpm !== 0) {
            writer.uint32(32).uint32(message.maintenanceFractionPpm);
        }
        if (!message.basePositionNotional.isZero()) {
            writer.uint32(40).uint64(message.basePositionNotional);
        }
        if (!message.openInterestLowerCap.isZero()) {
            writer.uint32(48).uint64(message.openInterestLowerCap);
        }
        if (!message.openInterestUpperCap.isZero()) {
            writer.uint32(56).uint64(message.openInterestUpperCap);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseLiquidityTierUpsertEventV2();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.id = reader.uint32();
                    break;
                case 2:
                    message.name = reader.string();
                    break;
                case 3:
                    message.initialMarginPpm = reader.uint32();
                    break;
                case 4:
                    message.maintenanceFractionPpm = reader.uint32();
                    break;
                case 5:
                    message.basePositionNotional = reader.uint64();
                    break;
                case 6:
                    message.openInterestLowerCap = reader.uint64();
                    break;
                case 7:
                    message.openInterestUpperCap = reader.uint64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c, _d;
        const message = createBaseLiquidityTierUpsertEventV2();
        message.id = (_a = object.id) !== null && _a !== void 0 ? _a : 0;
        message.name = (_b = object.name) !== null && _b !== void 0 ? _b : "";
        message.initialMarginPpm = (_c = object.initialMarginPpm) !== null && _c !== void 0 ? _c : 0;
        message.maintenanceFractionPpm = (_d = object.maintenanceFractionPpm) !== null && _d !== void 0 ? _d : 0;
        message.basePositionNotional = object.basePositionNotional !== undefined && object.basePositionNotional !== null ? helpers_1.Long.fromValue(object.basePositionNotional) : helpers_1.Long.UZERO;
        message.openInterestLowerCap = object.openInterestLowerCap !== undefined && object.openInterestLowerCap !== null ? helpers_1.Long.fromValue(object.openInterestLowerCap) : helpers_1.Long.UZERO;
        message.openInterestUpperCap = object.openInterestUpperCap !== undefined && object.openInterestUpperCap !== null ? helpers_1.Long.fromValue(object.openInterestUpperCap) : helpers_1.Long.UZERO;
        return message;
    }
};
function createBaseRegisterAffiliateEventV1() {
    return {
        referee: "",
        affiliate: ""
    };
}
exports.RegisterAffiliateEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.referee !== "") {
            writer.uint32(10).string(message.referee);
        }
        if (message.affiliate !== "") {
            writer.uint32(18).string(message.affiliate);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRegisterAffiliateEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.referee = reader.string();
                    break;
                case 2:
                    message.affiliate = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseRegisterAffiliateEventV1();
        message.referee = (_a = object.referee) !== null && _a !== void 0 ? _a : "";
        message.affiliate = (_b = object.affiliate) !== null && _b !== void 0 ? _b : "";
        return message;
    }
};
function createBaseUpsertVaultEventV1() {
    return {
        address: "",
        clobPairId: 0,
        status: 0
    };
}
exports.UpsertVaultEventV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        if (message.clobPairId !== 0) {
            writer.uint32(16).uint32(message.clobPairId);
        }
        if (message.status !== 0) {
            writer.uint32(24).int32(message.status);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpsertVaultEventV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.address = reader.string();
                    break;
                case 2:
                    message.clobPairId = reader.uint32();
                    break;
                case 3:
                    message.status = reader.int32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseUpsertVaultEventV1();
        message.address = (_a = object.address) !== null && _a !== void 0 ? _a : "";
        message.clobPairId = (_b = object.clobPairId) !== null && _b !== void 0 ? _b : 0;
        message.status = (_c = object.status) !== null && _c !== void 0 ? _c : 0;
        return message;
    }
};
