// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/listing/params.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListingVaultDepositParams = void 0;
const _m0 = __importStar(require("protobufjs/minimal"));
function createBaseListingVaultDepositParams() {
    return {
        newVaultDepositAmount: new Uint8Array(),
        mainVaultDepositAmount: new Uint8Array(),
        numBlocksToLockShares: 0
    };
}
exports.ListingVaultDepositParams = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.newVaultDepositAmount.length !== 0) {
            writer.uint32(10).bytes(message.newVaultDepositAmount);
        }
        if (message.mainVaultDepositAmount.length !== 0) {
            writer.uint32(18).bytes(message.mainVaultDepositAmount);
        }
        if (message.numBlocksToLockShares !== 0) {
            writer.uint32(24).uint32(message.numBlocksToLockShares);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseListingVaultDepositParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.newVaultDepositAmount = reader.bytes();
                    break;
                case 2:
                    message.mainVaultDepositAmount = reader.bytes();
                    break;
                case 3:
                    message.numBlocksToLockShares = reader.uint32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseListingVaultDepositParams();
        message.newVaultDepositAmount = (_a = object.newVaultDepositAmount) !== null && _a !== void 0 ? _a : new Uint8Array();
        message.mainVaultDepositAmount = (_b = object.mainVaultDepositAmount) !== null && _b !== void 0 ? _b : new Uint8Array();
        message.numBlocksToLockShares = (_c = object.numBlocksToLockShares) !== null && _c !== void 0 ? _c : 0;
        return message;
    }
};
