// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/listing/tx.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgUpgradeIsolatedPerpetualToCrossResponse = exports.MsgUpgradeIsolatedPerpetualToCross = exports.MsgSetListingVaultDepositParamsResponse = exports.MsgSetListingVaultDepositParams = exports.MsgCreateMarketPermissionlessResponse = exports.MsgCreateMarketPermissionless = exports.MsgSetMarketsHardCapResponse = exports.MsgSetMarketsHardCap = void 0;
const subaccount_1 = require("../subaccounts/subaccount");
const params_1 = require("./params");
const _m0 = __importStar(require("protobufjs/minimal"));
function createBaseMsgSetMarketsHardCap() {
    return {
        authority: "",
        hardCapForMarkets: 0
    };
}
exports.MsgSetMarketsHardCap = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.hardCapForMarkets !== 0) {
            writer.uint32(16).uint32(message.hardCapForMarkets);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetMarketsHardCap();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.hardCapForMarkets = reader.uint32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseMsgSetMarketsHardCap();
        message.authority = (_a = object.authority) !== null && _a !== void 0 ? _a : "";
        message.hardCapForMarkets = (_b = object.hardCapForMarkets) !== null && _b !== void 0 ? _b : 0;
        return message;
    }
};
function createBaseMsgSetMarketsHardCapResponse() {
    return {};
}
exports.MsgSetMarketsHardCapResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetMarketsHardCapResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgSetMarketsHardCapResponse();
        return message;
    }
};
function createBaseMsgCreateMarketPermissionless() {
    return {
        ticker: "",
        subaccountId: undefined
    };
}
exports.MsgCreateMarketPermissionless = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.ticker !== "") {
            writer.uint32(10).string(message.ticker);
        }
        if (message.subaccountId !== undefined) {
            subaccount_1.SubaccountId.encode(message.subaccountId, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgCreateMarketPermissionless();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.ticker = reader.string();
                    break;
                case 2:
                    message.subaccountId = subaccount_1.SubaccountId.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseMsgCreateMarketPermissionless();
        message.ticker = (_a = object.ticker) !== null && _a !== void 0 ? _a : "";
        message.subaccountId = object.subaccountId !== undefined && object.subaccountId !== null ? subaccount_1.SubaccountId.fromPartial(object.subaccountId) : undefined;
        return message;
    }
};
function createBaseMsgCreateMarketPermissionlessResponse() {
    return {};
}
exports.MsgCreateMarketPermissionlessResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgCreateMarketPermissionlessResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgCreateMarketPermissionlessResponse();
        return message;
    }
};
function createBaseMsgSetListingVaultDepositParams() {
    return {
        authority: "",
        params: undefined
    };
}
exports.MsgSetListingVaultDepositParams = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.params !== undefined) {
            params_1.ListingVaultDepositParams.encode(message.params, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetListingVaultDepositParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.params = params_1.ListingVaultDepositParams.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseMsgSetListingVaultDepositParams();
        message.authority = (_a = object.authority) !== null && _a !== void 0 ? _a : "";
        message.params = object.params !== undefined && object.params !== null ? params_1.ListingVaultDepositParams.fromPartial(object.params) : undefined;
        return message;
    }
};
function createBaseMsgSetListingVaultDepositParamsResponse() {
    return {};
}
exports.MsgSetListingVaultDepositParamsResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetListingVaultDepositParamsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgSetListingVaultDepositParamsResponse();
        return message;
    }
};
function createBaseMsgUpgradeIsolatedPerpetualToCross() {
    return {
        authority: "",
        perpetualId: 0
    };
}
exports.MsgUpgradeIsolatedPerpetualToCross = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.authority !== "") {
            writer.uint32(10).string(message.authority);
        }
        if (message.perpetualId !== 0) {
            writer.uint32(16).uint32(message.perpetualId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpgradeIsolatedPerpetualToCross();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.authority = reader.string();
                    break;
                case 2:
                    message.perpetualId = reader.uint32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseMsgUpgradeIsolatedPerpetualToCross();
        message.authority = (_a = object.authority) !== null && _a !== void 0 ? _a : "";
        message.perpetualId = (_b = object.perpetualId) !== null && _b !== void 0 ? _b : 0;
        return message;
    }
};
function createBaseMsgUpgradeIsolatedPerpetualToCrossResponse() {
    return {};
}
exports.MsgUpgradeIsolatedPerpetualToCrossResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgUpgradeIsolatedPerpetualToCrossResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgUpgradeIsolatedPerpetualToCrossResponse();
        return message;
    }
};
