// @dydxprotocol/v4-proto/src/codegen/dydxprotocol/listing/tx.rpc.msg.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgClientImpl = void 0;
const _m0 = __importStar(require("protobufjs/minimal"));
const tx_1 = require("./tx");
class MsgClientImpl {
    constructor(rpc) {
        this.rpc = rpc;
        this.setMarketsHardCap = this.setMarketsHardCap.bind(this);
        this.createMarketPermissionless = this.createMarketPermissionless.bind(this);
        this.setListingVaultDepositParams = this.setListingVaultDepositParams.bind(this);
        this.upgradeIsolatedPerpetualToCross = this.upgradeIsolatedPerpetualToCross.bind(this);
    }
    setMarketsHardCap(request) {
        const data = tx_1.MsgSetMarketsHardCap.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.listing.Msg", "SetMarketsHardCap", data);
        return promise.then(data => tx_1.MsgSetMarketsHardCapResponse.decode(new _m0.Reader(data)));
    }
    createMarketPermissionless(request) {
        const data = tx_1.MsgCreateMarketPermissionless.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.listing.Msg", "CreateMarketPermissionless", data);
        return promise.then(data => tx_1.MsgCreateMarketPermissionlessResponse.decode(new _m0.Reader(data)));
    }
    setListingVaultDepositParams(request) {
        const data = tx_1.MsgSetListingVaultDepositParams.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.listing.Msg", "SetListingVaultDepositParams", data);
        return promise.then(data => tx_1.MsgSetListingVaultDepositParamsResponse.decode(new _m0.Reader(data)));
    }
    upgradeIsolatedPerpetualToCross(request) {
        const data = tx_1.MsgUpgradeIsolatedPerpetualToCross.encode(request).finish();
        const promise = this.rpc.request("dydxprotocol.listing.Msg", "UpgradeIsolatedPerpetualToCross", data);
        return promise.then(data => tx_1.MsgUpgradeIsolatedPerpetualToCrossResponse.decode(new _m0.Reader(data)));
    }
}
exports.MsgClientImpl = MsgClientImpl;
